---
title: "Object Initialization and preprocessing"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Object Initialization and preprocessing}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
```{r, include = FALSE}


options(crayon.enabled = FALSE)
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
```

```{r, warning=FALSE, message=FALSE}
library(Rcongas)
library(dplyr)
library(ggplot2)
library(stringr)
```

In this Vignette we show how to create a CONGAS+ object starting from RNA and ATAC count matrices. We will use the 
B-cell lymphoma cells sequences via 10x genomics multiomic assay. The raw data can be foun [here](https://www.10xgenomics.com/resources/datasets/fresh-frozen-lymph-node-with-b-cell-lymphoma-14-k-sorted-nuclei-1-standard-2-0-0).
Setup the directory where results are stored.
```{r}

sample = 'lymphoma'

prefix = paste0(sample, "/tutorial/")

out.dir = paste0(prefix, "/congas_data/")
fig.dir = paste0(prefix, "/congas_figures/")

if (!dir.exists(out.dir)) {dir.create(out.dir, recursive = T)}
if (!dir.exists(fig.dir)) {dir.create(fig.dir, recursive = T)}
```

Load the ATAC counts to create the tibble needed as CONGAS+ input
```{r}

data(atac_counts) 
data(rna_counts)

```

With the multiome assay, the feature files provided by CellRanger contains already
a mapping between gene names and coordinates. Let's read this file
As an alternative, when such file not not readily available, users can exploit
BiomaRt. Please see the correspodning vignette \code{Gene mapping}

```{r}
data(features)
```


Create the RNA and ATAC tibble required by CONGAS+
```{r}
atac_featureDF = data.frame(id = rownames(atac_counts)) %>%
  tidyr::separate(id, c('chr', 'from', 'to'), '-')

atac = Rcongas::create_congas_tibble(counts = atac_counts, 
                                     modality = 'ATAC', 
                                     save_dir = NULL,
                                     features = atac_featureDF)

# Compute normalization factors
norm_atac = Rcongas:::auto_normalisation_factor(atac) %>%
  mutate(modality = 'ATAC')

rna = Rcongas::create_congas_tibble(counts = rna_counts, 
                                    modality = 'RNA', 
                                    save_dir=NULL, 
                                    features = features)


# Compute normalization factors
norm_rna = Rcongas:::auto_normalisation_factor(rna) %>%
  mutate(modality = 'RNA')

atac = atac %>% mutate(value = as.integer(value))
rna = rna %>% mutate(value = as.integer(value))
```

Apply some basic filters on known genes
```{R}
rna = rna %>% filter_known_genes(what='r')

all_genes = rna$gene %>% unique
mito = all_genes %>% str_starts(pattern = 'MT-')
all_genes = setdiff(all_genes, all_genes[mito])
rna = rna %>% dplyr::filter(gene %in% all_genes)
```

Read chromosome arm coordinated that will be used as segment breakpoints
```{R}
data(example_segments)

# Remove these chromosomes
segments = example_segments %>% dplyr::filter(chr != 'chrX', chr != 'chrY')
```

Now init the (R)CONGAS+ object. 
Users can choose between two object types: the multiomic or standard CONGAS+, setting the parameter \code{multiome} to TRUE or FALSE, respectively.
When the observations between RNA and ATAC are paired, the multiomic object should be used, in order to assign the RNA and ATAC observations from the same cells to the same clusters.
Please note that in this case the cells are expected to have the same barcode in the two modalities. The object provided as an example in this vignette illustrates this.
```{R}
x = init(
  rna = rna,
  atac = atac,
  segmentation = segments, #%>% mutate(copies = as.integer(round(copies))),
  rna_normalisation_factors = norm_rna,
  atac_normalisation_factors = norm_atac,
  rna_likelihood = "NB", 
  atac_likelihood = "NB",
  description = paste0('Bimodal ', sample),
  multiome = TRUE)
```

Now plot the number of nonzero cells vs the number of peaks per segment, to help 
identify a threshold for filtering segments. 

```{R}
ggplot(x$input$segmentation, aes(x=ATAC_nonzerocells, y=ATAC_peaks)) + geom_point() + theme_bw()
ggplot(x$input$segmentation, aes(x=RNA_nonzerocells, y=RNA_genes)) + geom_point() + theme_bw()
```

```{R}
x = Rcongas::filter_segments(x, RNA_nonzerocells = 300, ATAC_nonzerocells = 300)
```

Generate plots to check out the data distribution
```{r, fig.width=8, fig.height=8}
plot_data(
  x,
  what = 'histogram',
  segments = get_input(x, what = 'segmentation') %>%
    ungroup() %>%
    mutate(L = to - from) %>%
    dplyr::arrange(dplyr::desc(L)) %>%
    dplyr::top_n(9) %>%
    pull(segment_id)
)
```


Add metadata to annotate the counts distributions
```{r, fig.width=8, fig.height=8}
data(metadata)

x$input$metadata = metadata

plot_data(x, 
  to_plot = 'type', 
  position = 'stack',
  segments = get_input(x, what = 'segmentation') %>%
    ungroup() %>%
    mutate(L = to - from) %>%
    dplyr::arrange(dplyr::desc(L)) %>%
    dplyr::top_n(9) %>%
    pull(segment_id))
```

Please see the next vignette \code{Fitting} that shows how to perform CONGAS+ inference.