---
title: "Getter functions"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getter functions}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r, include = T, results='hide', warning=FALSE, message=FALSE}
require(Rcongas)
```

We show how you can access information from a CONGAS computation, using this package. Some functions might be only available for certain model objects, in that case errors are thrown.

In this vignette we use the default model (`MixtureGaussian`).

# Getters available in `rcongas`

We work with an example object released with the tool. 
```{r}
data('congas_example', package = 'Rcongas')

print(congas_example)
```

## Reference information

* `get_reference_genome`, `get_gene_annotations` to obtain information about the used reference.

The gene locations for the reference genomes supported by congas can be obtained by using `get_gene_annotations`. 

Note that the reference genome id is stored in the `rcongas` object created at the time the analysis is performed. So the genome coordinates of the genes are consistent with the adopted reference.

```{r, eval=FALSE}
# The adopted reference genome
get_reference_genome(congas_example)

# Gene coordinates as of the reference
get_gene_annotations(congas_example)
```

## Clustering results

* `get_k`, `get_clusters_size`, `get_clusters_ploidy` to obtain information about the output clusters;
* `get_clusters` to obtain clustering assignments.

You can obtain the value fit for $k\geq 1$, the number of clusters/ clones in the CONGAS model, using `get_k`. For every cluster, you can also obtain the total number of cells assigned to that, as well as its normalised value (i.e., the mixing proporitions).
```{r}
# Best number of clusters/ clones in the CONGAS model fit
get_k(congas_example)

# Cluster size
get_clusters_size(congas_example)

# Cluster size (normalised)
get_clusters_size(congas_example, normalise = TRUE)
```

Then you can also obtain the CNA profile estimated for each one of the $k$ clones, using `get_clusters_ploidy`. This reports, for every segment the lognormal CNA value estimated by CONGAS, as well as the segment mean, and the `CN` value (which is the offset to the mean of the segment). Values from this getter can be automatically filtered for certain chromsome ids, and clusters.
```{r}
# Get all data (all clones, all chromsomes)
get_clusters_ploidy(congas_example)

# Zoom in one specific chromosome
get_clusters_ploidy(congas_example, chromosomes = 'chr2')

# Zoom in one specific cluster, here "1"
get_clusters_ploidy(congas_example, clusters = '1')

# Zoom in one specific chromosome and cluster
get_clusters_ploidy(congas_example, chromosomes = 'chr2', clusters = "1")
```

Then you can also obtain the clustering assignments for each one of the input cells,  using function `get_clusters`. This getter can also subset cells in order  to contain only those that belong to a specific set of clusters.
```{r}
# Get all cells at once
get_clusters(congas_example)

# Subset to cluster "c1"
get_clusters(congas_example, clusters = 'c1')
```

## Stored input data

* `get_input_segmentation` to obtain information about the input DNA segmentation;
* `get_input_raw_data` and `get_counts` to obtain the raw input data (i.e., single-cell counts) and the counts per segment used by CONGAS.


The input DNA segmentation is stored in the result of a CONGAS run. You can access that with `get_input_segmentation` and create simple filters based on chromosome ids. 

Note that this tibble contains also columns that report the ploidy of the input segment (`ploidy_real`), the actual number of used genes retained that map to each segment (`mu`), and the total number of genes that map to the segments (`fixed_mu`). 

Genes mapped to segments are filtered as explained in `XXXX`. Column `segment_id` is just a key to identify univocally the input segments to CONGAS.


```{r}
# All the input segments
get_input_segmentation(congas_example)

# Segments in chromosomes 1 to 3 (ids: chr1, chr2, chr3)
get_input_segmentation(congas_example, chromosomes = paste0('chr', 1:3))
```


If you have run the main `rcongas` fit function `XXX` by using parameters `XXXX`, the raw input RNA data matrix have also been stored into the fit object (otherwise no, just to save space in the final object). 

You have a getter `get_input_raw_data` to obtain the input data when this has been stored in a `rcongas` object, or you need to load yourself the input data that you used to fit the model (in that case `get_input_raw_data` would return `NULL` and report a warning to screen). 
The raw input data are the counts, per gene (rows), per cell (columns), obtained from the scRNAseq experiment.

```{r}
# Input data - stored inside "congas_example" in this case
raw_input_gene_by_cell = get_input_raw_data(congas_example)

# Just the 2x2 submatrix (genes on the rows, cells on the columns)
# print(raw_input_gene_by_cell[1:2, 1:2])
```

The data processed by CONGAS are counts mapped to segments, which are stored in a matrix that contains cells on rows, and segments on columns. This matrix can be extracted in long tidy format; the actual counts value can be obtained before or after normalisation,  and can be subset to certain portions of the cancer genome. 

In this long view we report also the cluster id for each of the input cells. 

```{r}
# Normalised data (default)
get_counts(congas_example)

# A view on a certain chromosome 
get_counts(congas_example,
           chromosomes = 'chr3')

# Un-normalised data
get_counts(congas_example, normalise = FALSE)
```

If you want to know what genes have been used by CONGAS and to which segments these have been
mapped you can use function

```{r}
# All mapped genes
get_mapped_genes(congas_example)

# Genes mapped in chromosomes '1' and '5'
get_mapped_genes(congas_example, chromosomes = c('chr1', 'chr5'))
```

## Post-clustering analysis

* `get_DE_table` to obtain results from DE analysis;

Results from Differential Expression (DE) analysis using `Seurat` are available by using `get_DE_table`. By default there are returned sorted by adjusted p-value, and filters can be applied to subset by maximum adjusted p-value (`cut_pvalue`), absolute log-fold change (`cut_lfc`), and chromosome id. 

Note that the filters are applied jointly (i.e., as intersection).

```{r}
# Default results (sorted by adjusted p-value, default filters)
get_DE_table(congas_example)

# DE filtered by p-value cut at alpha-level 5%
get_DE_table(congas_example, cut_pvalue = 0.05)

# DE filtered to retain all values, regardless the LFC
get_DE_table(congas_example, cut_lfc = 0)

# A view on a certain chromosome 
get_DE_table(congas_example, chromosomes = 'chr15')
```

