---
title: "Rcongas"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Rcongas}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r, warning=FALSE, message=FALSE}
library(Rcongas)
library(dplyr)
```

# Analysis workflow

These data are required:

- single-cell RNA or ATAC sequencing data from independent cells, in the form of counts per cell;
- genome-wide segmentation with associated per-segment ploidy.

The formats of the inputs is discussed here.

The (R)CONGAS+ workflow consists in:

- (optional) applying [filters](1_create_congas_object.html) on the input data;
- creating an (R)CONGAS+ object;
- (optional) applying [filters](1_create_congas_object.html) on the mapped data;
- fitting a model

The package comes with input data to show what the required input formats are, and ready to use
objects for case studies that we have discussed in the main papers; see [References](../docs/reference/index.html) for details.

# Input formats

```{r}
# Example data available in the package
data('multiome_congas_object')
```

RNA and ATAC have the same input format reporting:

- `chr`, `from` and `to`: genomic locations of a gene (RNA) or a peak (ATAC)
- `cell`: a cell identifier
- `value`: the observed counts (discrete)
This tibble is created woth the function \code{create_congas_tibble}

Segments have this format:

- `chr`, `from` and `to`: genomic range of the segment
- `copies`: ploidy of each segment



# Creation of a new dataset

## multiomodal assay

Please see the [RCONGAS object creation vignette](1_create_congas_object.html) to understand how to create a RCONGAS+ object. 
Once it is successfully generated, we can visualize some information about the object

```{r}
multiome_congas_object %>% print
```


<!-- ```{r, fig.width=6, fig.height=6} -->
<!-- # Select the first 5 segments to plot -->
<!-- some_segments = get_input(x, what = "segmentation") %>%  -->
<!--   filter(row_number() < 5) %>%  -->
<!--   pull(segment_id) -->

<!-- plot_data(x, what = 'histogram', segments = some_segments) -->
<!-- ``` -->

<!-- ```{r, fig.width=8, fig.height=4} -->
<!-- plot_data(x, what = 'lineplot') -->
<!-- ``` -->

<!-- ```{r, fig.width=6, fig.height=6} -->
<!-- plot_data(x, what = 'heatmap') -->
<!-- ``` -->


<!-- 
# Datasets

## RNA

```{r}
# Breast-cancer PDX
data('campbell_bcpdx')
campbell_bcpdx %>% print
```

```{r}
# Glioblastoma tumour normal
data('patel_gbmtn')
patel_gbmtn %>% print
```

```{r}
# Hematological bone marrow failure
data('zaho_hemato')
zaho_hemato %>% print
``` 
-->


