<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Full inference tutorial • Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Full inference tutorial">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Tutorial.html">Tutorial</a>
    </li>
    <li>
      <a href="../articles/3_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/4_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/5_Preprocessing.html">Preprocessing</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/militeee/Rcongas/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Full inference tutorial</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/militeee/Rcongasvignettes/2_Tutorial.Rmd" class="external-link"><code>vignettes/2_Tutorial.Rmd</code></a></small>
      <div class="hidden name"><code>2_Tutorial.Rmd</code></div>

    </div>

    
    
<p>In this vignette we will show a end-to-end example of rcongas on the
PDX dataset used in the papers.</p>
<div class="section level2">
<h2 id="data-loading-and-visualization">Data loading and visualization<a class="anchor" aria-label="anchor" href="#data-loading-and-visualization"></a>
</h2>
<p>We start with the two main inputs of Rcongas: a CNV data.frame and a
count matrix</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">Rcongas</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">rehsape2</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>

<span class="co"># required for using as an input sparse matrices</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org/" class="external-link">Matrix</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://github.com/const-ae/sparseMatrixStats" class="external-link">sparseMatrixStats</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"congas_example"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># genes by cells matrix</span>
<span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="va">congas_example</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">raw</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/cast.html" class="external-link">acast</a></span><span class="op">(</span><span class="va">gene</span> <span class="op">~</span> <span class="va">cell</span>, value.var <span class="op">=</span> <span class="st">"n"</span>, fill <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>

<span class="co"># or do some preprocessing like in Seurat, the functions get the matrix in Seurat format so we have to transpose</span>
<span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="fu">Rcongas</span><span class="fu">:::</span><span class="fu">preprocess_sc</span><span class="op">(</span><span class="va">count_matrix</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">t</span>,filter_upper_quantile <span class="op">=</span> <span class="cn">T</span>, perc_cells_gene_expr <span class="op">=</span> <span class="fl">0.05</span>,upper_quantile <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">t</span></code></pre></div>
<pre><code><span class="co">## Filtering cells</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> removed 1966 cells from a total of 2470</span></code></pre>
<pre><code><span class="co">## Filtering genes</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> removed 11056 genes from a total of 21247</span></code></pre>
<pre><code><span class="co">## Filtering genes on quantile</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> removed 510 genes from a total of 10191</span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># the cnv_table has 4 mandatory coloumns chr,from,to and tot (bulk CNV value)</span>
<span class="va">cnv_table</span> <span class="op">&lt;-</span> <span class="va">congas_example</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">cnv</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">chr</span>,<span class="va">from</span>,<span class="va">to</span>,<span class="va">tot</span><span class="op">)</span>


<span class="va">cnv_table</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #949494;"># A tibble: 5 × 4</span></span>
<span class="co">##   chr        from        to   tot</span>
<span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">## <span style="color: #BCBCBC;">1</span> chr1          1  12<span style="text-decoration: underline;">600</span>000     1</span>
<span class="co">## <span style="color: #BCBCBC;">2</span> chr1   12<span style="text-decoration: underline;">600</span>001  60<span style="text-decoration: underline;">150</span>000     2</span>
<span class="co">## <span style="color: #BCBCBC;">3</span> chr1   61<span style="text-decoration: underline;">200</span>001  77<span style="text-decoration: underline;">100</span>000     2</span>
<span class="co">## <span style="color: #BCBCBC;">4</span> chr1   78<span style="text-decoration: underline;">750</span>001 156<span style="text-decoration: underline;">000</span>000     2</span>
<span class="co">## <span style="color: #BCBCBC;">5</span> chr1  156<span style="text-decoration: underline;">000</span>001 249<span style="text-decoration: underline;">300</span>000     3</span></code></pre>
<p>Then we use the function <code>init</code> to construct a new Rcongas
object</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># You might try with and without correct_bins which remove some small segments and them merge </span>
<span class="co"># fragmented pieces with the same CN, as it sometimes can include some spurois correlations</span>

<span class="va">input_rcongas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">count_matrix</span> , cnv_data <span class="op">=</span> <span class="va">cnv_table</span>, description <span class="op">=</span> <span class="st">"congas_example"</span>, reference_genome <span class="op">=</span> <span class="st">"hg19"</span>, online <span class="op">=</span> <span class="cn">FALSE</span>, correct_bins <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Extracting gene data from reference.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> Validated input(s): <span style="color: #00BB00;">504</span> cells, <span style="color: #00BB00;">9681</span> genes and <span style="color: #00BB00;">70</span> CNA segments.</span></code></pre>
<pre><code><span class="co">## Loading required package: biomaRt</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Processing input counts.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Assembly Rcongas object.</span></code></pre>
<pre><code><span class="co">## <span style="color: #BBBB00;">!</span> Mapping inconsistent for <span style="color: #00BB00;">345</span> genes out <span style="color: #00BB00;">8571</span>, removing those from the raw data table.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Retaining <span style="color: #0000BB;">35.6 Mb</span> long-format tibble data with <span style="color: #00BB00;">1531708</span> points, matrix was <span style="color: #BB0000;">37.9 Mb</span>.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Object size in memory: <span style="color: #0000BB;">36.8 Mb</span>.</span></code></pre>
<pre><code><span class="co">## ── <span style="color: #000000; background-color: #BBBB00;">[ Rcongas ]</span><span style="color: #0000BB;"> congas_example</span> ──────────────────────────────────────────────────</span></code></pre>
<pre><code><span class="co">## → Data: <span style="color: #00BB00;">504</span> cells with <span style="color: #00BB00;">8199</span> genes, aggregated in <span style="color: #00BB00;">70</span> segments.</span></code></pre>
<pre><code><span class="co">## <span style="color: #BBBB00;">!</span> Clusters: <span style="color: #BB0000;">not available</span>.</span></code></pre>
<p>We can have a look at the distribution of the number of genes in the
segments and filter accordingly</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">input_rcongas</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">cnv</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">mu</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">50</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">10</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span>, color <span class="op">=</span> <span class="st">"indianred"</span>, size <span class="op">=</span> <span class="fl">1.5</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="2_Tutorial_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">new_cnv_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_segments_and_return_cnv_table.html">filter_segments_and_return_cnv_table</a></span><span class="op">(</span><span class="va">input_rcongas</span>,mu <span class="op">=</span>  <span class="fl">10</span>, length <span class="op">=</span> <span class="fl">0</span>, merge <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">input_rcongas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/init.html">init</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">count_matrix</span>, cnv_data <span class="op">=</span> <span class="va">new_cnv_table</span>, description <span class="op">=</span> <span class="st">"congas_example"</span>, reference_genome <span class="op">=</span> <span class="st">"hg19"</span>, online <span class="op">=</span> <span class="cn">FALSE</span>, correct_bins <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Extracting gene data from reference.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> Validated input(s): <span style="color: #00BB00;">504</span> cells, <span style="color: #00BB00;">9681</span> genes and <span style="color: #00BB00;">56</span> CNA segments.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Processing input counts.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Assembly Rcongas object.</span></code></pre>
<pre><code><span class="co">## <span style="color: #BBBB00;">!</span> Mapping inconsistent for <span style="color: #00BB00;">338</span> genes out <span style="color: #00BB00;">8579</span>, removing those from the raw data table.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Retaining <span style="color: #0000BB;">35.6 Mb</span> long-format tibble data with <span style="color: #00BB00;">1531708</span> points, matrix was <span style="color: #BB0000;">37.9 Mb</span>.</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">ℹ</span> Object size in memory: <span style="color: #0000BB;">36.8 Mb</span>.</span></code></pre>
<pre><code><span class="co">## ── <span style="color: #000000; background-color: #BBBB00;">[ Rcongas ]</span><span style="color: #0000BB;"> congas_example</span> ──────────────────────────────────────────────────</span></code></pre>
<pre><code><span class="co">## → Data: <span style="color: #00BB00;">504</span> cells with <span style="color: #00BB00;">8215</span> genes, aggregated in <span style="color: #00BB00;">56</span> segments.</span></code></pre>
<pre><code><span class="co">## <span style="color: #BBBB00;">!</span> Clusters: <span style="color: #BB0000;">not available</span>.</span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># automatic tuning of library size</span>

<span class="va">theta_vals</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/approx_theta_prior_params.html">approx_theta_prior_params</a></span><span class="op">(</span><span class="va">input_rcongas</span>, plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="2_Tutorial_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># in general it is wise to set normalize_by_segs = TRUE when you have more than 1000/1200 cells to avoid overfitting</span>
<span class="co"># Try method = BIC for a less conservative clustering</span>
<span class="va">inference</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/best_cluster.html">best_cluster</a></span><span class="op">(</span><span class="va">input_rcongas</span>, clusters <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, model <span class="op">=</span> <span class="st">"MixtureGaussian"</span>, param_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>theta_shape <span class="op">=</span> <span class="va">theta_vals</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, theta_rate <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>, 
                          steps <span class="op">=</span> <span class="fl">500</span>, lr <span class="op">=</span> <span class="fl">0.01</span>, MAP <span class="op">=</span> <span class="cn">TRUE</span>, seed <span class="op">=</span> <span class="fl">3</span>, normalize_by_segs <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"ICL"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## <span style="color: #BBBB00;">!</span> Removing clusters collapsed to 0 probability!</span></code></pre>
<pre><code><span class="co">## [1] "Best number of cluster is 4"</span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_counts_rna_segments.html">plot_counts_rna_segments</a></span><span class="op">(</span><span class="va">inference</span><span class="op">)</span></code></pre></div>
<p><img src="2_Tutorial_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>It is especially important to look at the actual distributions as
CONGAS my have a slight tendency to overfit, especially when dealing
with very heterogeneous datasets or when there is high level of variance
in the library size factors.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'gridExtra' was built under R version 4.1.1</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## Attaching package: 'gridExtra'</span></code></pre>
<pre><code><span class="co">## The following object is masked from 'package:dplyr':</span>
<span class="co">## </span>
<span class="co">##     combine</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">high_segs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/highlights.html">highlights</a></span><span class="op">(</span><span class="va">inference</span>, alpha <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">highlight</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">segment_id</span><span class="op">)</span>

<span class="va">segs_p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_segment_density.html">plot_segment_density</a></span><span class="op">(</span><span class="va">inference</span>, segments_ids <span class="op">=</span> <span class="va">high_segs</span>, sum_denominator <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">## "none")` instead.</span></code></pre>
<pre><code><span class="co">## Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">## "none")` instead.</span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="st">"grid.arrange"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">segs_p</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="2_Tutorial_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>We can also automatically calculate the differential expression
between cluster 1 and 2. In order to do so we have to pass both the
fitted object and the count matrix to the <code>calculate_DE</code>
function. The underlying implementation is based on
<code>Seurat</code>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">de_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculate_DE.html">calculate_DE</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">inference</span>, input <span class="op">=</span> <span class="va">count_matrix</span>, clone1 <span class="op">=</span> <span class="st">"c1"</span>, clone2 <span class="op">=</span> <span class="st">"c2"</span>, method <span class="op">=</span> <span class="st">"wilcox"</span>, logfc.threshold <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span></code></pre>
<pre><code><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Differential Expression analysis with Seurat</span> <span style="color: #00BBBB;">────────────────────────────────</span></span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">## <span style="color: #000000;">                      ┌────────────────────────────────────┐</span></span>
<span class="co">##                       <span style="color: #000000;">│</span><span style="background-color: #FFAF00;">                                    </span><span style="color: #000000;">│</span></span>
<span class="co">##                       <span style="color: #000000;">│</span><span style="background-color: #FFAF00;">   </span><span style="color: #000000; background-color: #FFAF00;">Groups: c1 vs c2; using wilcox</span><span style="background-color: #FFAF00;">   </span><span style="color: #000000;">│</span></span>
<span class="co">##                       <span style="color: #000000;">│</span><span style="background-color: #FFAF00;">                                    </span><span style="color: #000000;">│</span></span>
<span class="co">## <span style="color: #000000;">                      └────────────────────────────────────┘</span></span></code></pre>
<pre><code><span class="co">## - With normalisation: <span style="color: #00BB00;">TRUE</span>; LFC threshold <span style="color: #00BB00;">0.5</span>.</span></code></pre>
<pre><code><span class="co">## Loading required package: Seurat</span></code></pre>
<pre><code><span class="co">## Warning: package 'Seurat' was built under R version 4.1.1</span></code></pre>
<pre><code><span class="co">## Attaching SeuratObject</span></code></pre>
<pre><code><span class="co">## <span style="color: #00BB00;">✓</span> Found <span style="color: #00BB00;">188</span> EG at alpha level 0.05.</span></code></pre>
<p>The function returns a Rcongas object with a new field
<code>$DE</code> which stores the parameters used to perform the DE and
the table of differentially expressed genes with adjusted p values (BH)
and log2FC.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_DE_volcano.html">plot_DE_volcano</a></span><span class="op">(</span><span class="va">de_object</span><span class="op">)</span></code></pre></div>
<p><img src="2_Tutorial_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://sites.google.com/view/cdslab" class="external-link">Salvatore
Milite</a>, <a href="https://sites.google.com/view/cdslab" class="external-link">Giulio
Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
