<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getter functions • Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getter functions">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Tutorial.html">Tutorial</a>
    </li>
    <li>
      <a href="../articles/3_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/4_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/5_Preprocessing.html">Preprocessing</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/militeee/Rcongas/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getter functions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/militeee/Rcongasvignettes/3_Getters.Rmd" class="external-link"><code>vignettes/3_Getters.Rmd</code></a></small>
      <div class="hidden name"><code>3_Getters.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">Rcongas</span><span class="op">)</span></code></pre></div>
<p>We show how you can access information from a CONGAS computation,
using this package. Some functions might be only available for certain
model objects, in that case errors are thrown.</p>
<p>In this vignette we use the default model
(<code>MixtureGaussian</code>).</p>
<div class="section level2">
<h2 id="getters-available-in-rcongas">Getters available in <code>rcongas</code><a class="anchor" aria-label="anchor" href="#getters-available-in-rcongas"></a>
</h2>
<p>We work with an example object released with the tool.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'congas_example'</span>, package <span class="op">=</span> <span class="st">'Rcongas'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ Rcongas ]</span><span style="color: #0000BB;"> </span> ────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; → Data: <span style="color: #00BB00;">503</span> cells with <span style="color: #00BB00;">8564</span> genes, aggregated in <span style="color: #00BB00;">70</span> segments.</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Clusters: <span style="color: #00BB00;">k = 2</span>, model with <span style="color: #00BB00;">AIC</span> = 577427.35.</span>
<span class="co">#&gt; ● Cluster <span style="color: #00BB00;">c1</span>, n = 379 [75.35% of total cells].</span>
<span class="co">#&gt; ● Cluster <span style="color: #00BB00;">c2</span>, n = 124 [24.65% of total cells].</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── CNA highlights (alpha = 0.05)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✓</span>  <span style="color: #00BB00;">c1</span> CNA(s):  <span style="color: #00BB00;">[chr14:93300001:105450000] vs c2</span>, <span style="color: #00BB00;">[chr15:67050001:102600000] vs c2</span>, <span style="color: #00BB00;">[chr16:1:3750000] vs c2</span>, and <span style="color: #00BB00;">[chr18:32400001:55950000] vs c2</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Differential Expression <span style="color: #00BB00;">DESeq2</span>  (<span style="color: #00BB00;">1</span> vs <span style="color: #00BB00;">2</span>)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✓</span> <span style="color: #00BB00;">212</span> DE genes (alpha = 0.01, |lfc| &gt; 0.25).</span></code></pre></div>
<div class="section level3">
<h3 id="reference-information">Reference information<a class="anchor" aria-label="anchor" href="#reference-information"></a>
</h3>
<ul>
<li>
<code>get_reference_genome</code>, <code>get_gene_annotations</code>
to obtain information about the used reference.</li>
</ul>
<p>The gene locations for the reference genomes supported by congas can
be obtained by using <code>get_gene_annotations</code>.</p>
<p>Note that the reference genome id is stored in the
<code>rcongas</code> object created at the time the analysis is
performed. So the genome coordinates of the genes are consistent with
the adopted reference.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The adopted reference genome</span>
<span class="fu"><a href="../reference/get_reference_genome.html">get_reference_genome</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>

<span class="co"># Gene coordinates as of the reference</span>
<span class="fu"><a href="../reference/get_gene_annotations.html">get_gene_annotations</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="clustering-results">Clustering results<a class="anchor" aria-label="anchor" href="#clustering-results"></a>
</h3>
<ul>
<li>
<code>get_k</code>, <code>get_clusters_size</code>,
<code>get_clusters_ploidy</code> to obtain information about the output
clusters;</li>
<li>
<code>get_clusters</code> to obtain clustering assignments.</li>
</ul>
<p>You can obtain the value fit for <span class="math inline">\(k\geq
1\)</span>, the number of clusters/ clones in the CONGAS model, using
<code>get_k</code>. For every cluster, you can also obtain the total
number of cells assigned to that, as well as its normalised value (i.e.,
the mixing proporitions).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Best number of clusters/ clones in the CONGAS model fit</span>
<span class="fu"><a href="../reference/get_k.html">get_k</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; [1] 2</span>

<span class="co"># Cluster size</span>
<span class="fu"><a href="../reference/get_clusters_size.html">get_clusters_size</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt;  c1  c2 </span>
<span class="co">#&gt; 379 124</span>

<span class="co"># Cluster size (normalised)</span>
<span class="fu"><a href="../reference/get_clusters_size.html">get_clusters_size</a></span><span class="op">(</span><span class="va">congas_example</span>, normalise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt;        c1        c2 </span>
<span class="co">#&gt; 0.7534791 0.2465209</span></code></pre></div>
<p>Then you can also obtain the CNA profile estimated for each one of
the <span class="math inline">\(k\)</span> clones, using
<code>get_clusters_ploidy</code>. This reports, for every segment the
lognormal CNA value estimated by CONGAS, as well as the segment mean,
and the <code>CN</code> value (which is the offset to the mean of the
segment). Values from this getter can be automatically filtered for
certain chromsome ids, and clusters.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get all data (all clones, all chromsomes)</span>
<span class="fu"><a href="../reference/get_clusters_ploidy.html">get_clusters_ploidy</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 140 × 9</span></span>
<span class="co">#&gt;    chr        from        to cluster       CN segment_mean offset_CN segment_id </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1          1  12<span style="text-decoration: underline;">600</span>000 c1       0.043<span style="text-decoration: underline;">6</span>          1.20   0.043<span style="text-decoration: underline;">6</span>  chr1:1:126…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   12<span style="text-decoration: underline;">600</span>001  60<span style="text-decoration: underline;">150</span>000 c1       0.016<span style="text-decoration: underline;">5</span>          1.58   0.016<span style="text-decoration: underline;">5</span>  chr1:12600…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   61<span style="text-decoration: underline;">200</span>001  77<span style="text-decoration: underline;">100</span>000 c1      -<span style="color: #BB0000;">0.042</span><span style="color: #BB0000; text-decoration: underline;">7</span>          1.32  -<span style="color: #BB0000;">0.042</span><span style="color: #BB0000; text-decoration: underline;">7</span>  chr1:61200…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   78<span style="text-decoration: underline;">750</span>001 156<span style="text-decoration: underline;">000</span>000 c1       0.018<span style="text-decoration: underline;">4</span>          1.59   0.018<span style="text-decoration: underline;">4</span>  chr1:78750…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  156<span style="text-decoration: underline;">000</span>001 249<span style="text-decoration: underline;">300</span>000 c1       0.009<span style="text-decoration: underline;">60</span>         1.51   0.009<span style="text-decoration: underline;">60</span> chr1:15600…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr2          1  26<span style="text-decoration: underline;">250</span>000 c1       0.079<span style="text-decoration: underline;">7</span>          1.62   0.079<span style="text-decoration: underline;">7</span>  chr2:1:262…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2   26<span style="text-decoration: underline;">250</span>001  38<span style="text-decoration: underline;">100</span>000 c1       0.013<span style="text-decoration: underline;">9</span>          1.31   0.013<span style="text-decoration: underline;">9</span>  chr2:26250…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2   64<span style="text-decoration: underline;">050</span>001  87<span style="text-decoration: underline;">150</span>000 c1      -<span style="color: #BB0000;">0.006</span><span style="color: #BB0000; text-decoration: underline;">52</span>         1.49  -<span style="color: #BB0000;">0.006</span><span style="color: #BB0000; text-decoration: underline;">52</span> chr2:64050…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2   88<span style="text-decoration: underline;">350</span>001 159<span style="text-decoration: underline;">900</span>000 c1      -<span style="color: #BB0000;">0.041</span><span style="color: #BB0000; text-decoration: underline;">8</span>          1.52  -<span style="color: #BB0000;">0.041</span><span style="color: #BB0000; text-decoration: underline;">8</span>  chr2:88350…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr2  159<span style="text-decoration: underline;">900</span>001 176<span style="text-decoration: underline;">850</span>000 c1       0.024<span style="text-decoration: underline;">7</span>          1.24   0.024<span style="text-decoration: underline;">7</span>  chr2:15990…</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 130 more rows, and 1 more variable: highlight &lt;lgl&gt;</span></span>

<span class="co"># Zoom in one specific chromosome</span>
<span class="fu"><a href="../reference/get_clusters_ploidy.html">get_clusters_ploidy</a></span><span class="op">(</span><span class="va">congas_example</span>, chromosomes <span class="op">=</span> <span class="st">'chr2'</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 14 × 9</span></span>
<span class="co">#&gt;    chr        from        to cluster       CN segment_mean offset_CN segment_id </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr2          1  26<span style="text-decoration: underline;">250</span>000 c1       0.079<span style="text-decoration: underline;">7</span>          1.62   0.079<span style="text-decoration: underline;">7</span>  chr2:1:262…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr2   26<span style="text-decoration: underline;">250</span>001  38<span style="text-decoration: underline;">100</span>000 c1       0.013<span style="text-decoration: underline;">9</span>          1.31   0.013<span style="text-decoration: underline;">9</span>  chr2:26250…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr2   64<span style="text-decoration: underline;">050</span>001  87<span style="text-decoration: underline;">150</span>000 c1      -<span style="color: #BB0000;">0.006</span><span style="color: #BB0000; text-decoration: underline;">52</span>         1.49  -<span style="color: #BB0000;">0.006</span><span style="color: #BB0000; text-decoration: underline;">52</span> chr2:64050…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr2   88<span style="text-decoration: underline;">350</span>001 159<span style="text-decoration: underline;">900</span>000 c1      -<span style="color: #BB0000;">0.041</span><span style="color: #BB0000; text-decoration: underline;">8</span>          1.52  -<span style="color: #BB0000;">0.041</span><span style="color: #BB0000; text-decoration: underline;">8</span>  chr2:88350…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr2  159<span style="text-decoration: underline;">900</span>001 176<span style="text-decoration: underline;">850</span>000 c1       0.024<span style="text-decoration: underline;">7</span>          1.24   0.024<span style="text-decoration: underline;">7</span>  chr2:15990…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr2  176<span style="text-decoration: underline;">850</span>001 195<span style="text-decoration: underline;">450</span>000 c1      -<span style="color: #BB0000;">0.016</span><span style="color: #BB0000; text-decoration: underline;">8</span>          1.47  -<span style="color: #BB0000;">0.016</span><span style="color: #BB0000; text-decoration: underline;">8</span>  chr2:17685…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2  201<span style="text-decoration: underline;">750</span>001 243<span style="text-decoration: underline;">300</span>000 c1       0.009<span style="text-decoration: underline;">15</span>         1.42   0.009<span style="text-decoration: underline;">15</span> chr2:20175…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2          1  26<span style="text-decoration: underline;">250</span>000 c2      -<span style="color: #BB0000;">0.079</span><span style="color: #BB0000; text-decoration: underline;">7</span>          1.62  -<span style="color: #BB0000;">0.079</span><span style="color: #BB0000; text-decoration: underline;">7</span>  chr2:1:262…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2   26<span style="text-decoration: underline;">250</span>001  38<span style="text-decoration: underline;">100</span>000 c2      -<span style="color: #BB0000;">0.013</span><span style="color: #BB0000; text-decoration: underline;">9</span>          1.31  -<span style="color: #BB0000;">0.013</span><span style="color: #BB0000; text-decoration: underline;">9</span>  chr2:26250…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr2   64<span style="text-decoration: underline;">050</span>001  87<span style="text-decoration: underline;">150</span>000 c2       0.006<span style="text-decoration: underline;">52</span>         1.49   0.006<span style="text-decoration: underline;">52</span> chr2:64050…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr2   88<span style="text-decoration: underline;">350</span>001 159<span style="text-decoration: underline;">900</span>000 c2       0.041<span style="text-decoration: underline;">8</span>          1.52   0.041<span style="text-decoration: underline;">8</span>  chr2:88350…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr2  159<span style="text-decoration: underline;">900</span>001 176<span style="text-decoration: underline;">850</span>000 c2      -<span style="color: #BB0000;">0.024</span><span style="color: #BB0000; text-decoration: underline;">7</span>          1.24  -<span style="color: #BB0000;">0.024</span><span style="color: #BB0000; text-decoration: underline;">7</span>  chr2:15990…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr2  176<span style="text-decoration: underline;">850</span>001 195<span style="text-decoration: underline;">450</span>000 c2       0.016<span style="text-decoration: underline;">8</span>          1.47   0.016<span style="text-decoration: underline;">8</span>  chr2:17685…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr2  201<span style="text-decoration: underline;">750</span>001 243<span style="text-decoration: underline;">300</span>000 c2      -<span style="color: #BB0000;">0.009</span><span style="color: #BB0000; text-decoration: underline;">15</span>         1.42  -<span style="color: #BB0000;">0.009</span><span style="color: #BB0000; text-decoration: underline;">15</span> chr2:20175…</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 1 more variable: highlight &lt;lgl&gt;</span></span>

<span class="co"># Zoom in one specific cluster, here "1"</span>
<span class="fu"><a href="../reference/get_clusters_ploidy.html">get_clusters_ploidy</a></span><span class="op">(</span><span class="va">congas_example</span>, clusters <span class="op">=</span> <span class="st">'1'</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 9</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 9 variables: chr &lt;chr&gt;, from &lt;dbl&gt;, to &lt;dbl&gt;, cluster &lt;chr&gt;, CN &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   segment_mean &lt;dbl&gt;, offset_CN &lt;dbl&gt;, segment_id &lt;chr&gt;, highlight &lt;lgl&gt;</span></span>

<span class="co"># Zoom in one specific chromosome and cluster</span>
<span class="fu"><a href="../reference/get_clusters_ploidy.html">get_clusters_ploidy</a></span><span class="op">(</span><span class="va">congas_example</span>, chromosomes <span class="op">=</span> <span class="st">'chr2'</span>, clusters <span class="op">=</span> <span class="st">"1"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 0 × 9</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 9 variables: chr &lt;chr&gt;, from &lt;dbl&gt;, to &lt;dbl&gt;, cluster &lt;chr&gt;, CN &lt;dbl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   segment_mean &lt;dbl&gt;, offset_CN &lt;dbl&gt;, segment_id &lt;chr&gt;, highlight &lt;lgl&gt;</span></span></code></pre></div>
<p>Then you can also obtain the clustering assignments for each one of
the input cells, using function <code>get_clusters</code>. This getter
can also subset cells in order to contain only those that belong to a
specific set of clusters.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Get all cells at once</span>
<span class="fu"><a href="../reference/get_clusters.html">get_clusters</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 503 × 5</span></span>
<span class="co">#&gt;    cell               cluster    c1    c2 p_assignment</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> AAACCTGTCGACAGCC-1 c2          1     0            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> AAAGATGGTAATAGCA-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> AACACGTAGCTATGCT-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> AACTCCCAGACAGGCT-1 c2          1     0            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> AACTCCCGTACACCGC-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> AACTGGTCAAAGAATC-1 c2          1     0            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> AACTGGTCAAAGTGCG-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> AACTTTCGTGGCTCCA-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> AACTTTCTCGGTTAAC-1 c2          1     0            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> AAGACCTAGCCAGTAG-1 c2          1     0            1</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 493 more rows</span></span>

<span class="co"># Subset to cluster "c1"</span>
<span class="fu"><a href="../reference/get_clusters.html">get_clusters</a></span><span class="op">(</span><span class="va">congas_example</span>, clusters <span class="op">=</span> <span class="st">'c1'</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 379 × 5</span></span>
<span class="co">#&gt;    cell               cluster    c1    c2 p_assignment</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> AAAGATGGTAATAGCA-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> AACACGTAGCTATGCT-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> AACTCCCGTACACCGC-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> AACTGGTCAAAGTGCG-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> AACTTTCGTGGCTCCA-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> AAGCCGCGTTGTACAC-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> AATCCAGCATTTCAGG-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> AATCCAGTCCGAACGC-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> AATCGGTAGTGTTGAA-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> AATCGGTTCTGTCTAT-1 c1          0     1            1</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 369 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="stored-input-data">Stored input data<a class="anchor" aria-label="anchor" href="#stored-input-data"></a>
</h3>
<ul>
<li>
<code>get_input_segmentation</code> to obtain information about the
input DNA segmentation;</li>
<li>
<code>get_input_raw_data</code> and <code>get_counts</code> to
obtain the raw input data (i.e., single-cell counts) and the counts per
segment used by CONGAS.</li>
</ul>
<p>The input DNA segmentation is stored in the result of a CONGAS run.
You can access that with <code>get_input_segmentation</code> and create
simple filters based on chromosome ids.</p>
<p>Note that this tibble contains also columns that report the ploidy of
the input segment (<code>ploidy_real</code>), the actual number of used
genes retained that map to each segment (<code>mu</code>), and the total
number of genes that map to the segments (<code>fixed_mu</code>).</p>
<p>Genes mapped to segments are filtered as explained in
<code>XXXX</code>. Column <code>segment_id</code> is just a key to
identify univocally the input segments to CONGAS.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># All the input segments</span>
<span class="fu"><a href="../reference/get_input_segmentation.html">get_input_segmentation</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 70 × 12</span></span>
<span class="co">#&gt;    chr       start     end ploidy_real   tot   from     to   dist    mu fixed_mu</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1          1  1.26<span style="color: #949494;">e</span>7           1     1 1   <span style="color: #949494;">e</span>0 1.26<span style="color: #949494;">e</span>7 1.26<span style="color: #949494;">e</span>7    16       73</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   12<span style="text-decoration: underline;">600</span>001  6.02<span style="color: #949494;">e</span>7           2     2 1.26<span style="color: #949494;">e</span>7 6.02<span style="color: #949494;">e</span>7 4.75<span style="color: #949494;">e</span>7   102      302</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   61<span style="text-decoration: underline;">200</span>001  7.71<span style="color: #949494;">e</span>7           2     2 6.12<span style="color: #949494;">e</span>7 7.71<span style="color: #949494;">e</span>7 1.59<span style="color: #949494;">e</span>7    11       32</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   78<span style="text-decoration: underline;">750</span>001  1.56<span style="color: #949494;">e</span>8           2     2 7.88<span style="color: #949494;">e</span>7 1.56<span style="color: #949494;">e</span>8 7.72<span style="color: #949494;">e</span>7    87      261</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  156<span style="text-decoration: underline;">000</span>001  2.49<span style="color: #949494;">e</span>8           3     3 1.56<span style="color: #949494;">e</span>8 2.49<span style="color: #949494;">e</span>8 9.33<span style="color: #949494;">e</span>7   116      350</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr2          1  2.62<span style="color: #949494;">e</span>7           1     1 1   <span style="color: #949494;">e</span>0 2.62<span style="color: #949494;">e</span>7 2.62<span style="color: #949494;">e</span>7    15       56</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2   26<span style="text-decoration: underline;">250</span>001  3.81<span style="color: #949494;">e</span>7           2     2 2.63<span style="color: #949494;">e</span>7 3.81<span style="color: #949494;">e</span>7 1.18<span style="color: #949494;">e</span>7    19       59</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2   64<span style="text-decoration: underline;">050</span>001  8.72<span style="color: #949494;">e</span>7           2     2 6.41<span style="color: #949494;">e</span>7 8.72<span style="color: #949494;">e</span>7 2.31<span style="color: #949494;">e</span>7    35       95</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2   88<span style="text-decoration: underline;">350</span>001  1.60<span style="color: #949494;">e</span>8           2     2 8.84<span style="color: #949494;">e</span>7 1.60<span style="color: #949494;">e</span>8 7.15<span style="color: #949494;">e</span>7    48      159</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr2  159<span style="text-decoration: underline;">900</span>001  1.77<span style="color: #949494;">e</span>8           1     1 1.60<span style="color: #949494;">e</span>8 1.77<span style="color: #949494;">e</span>8 1.69<span style="color: #949494;">e</span>7    11       35</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 60 more rows, and 2 more variables: segment_id &lt;chr&gt;, size &lt;int&gt;</span></span>

<span class="co"># Segments in chromosomes 1 to 3 (ids: chr1, chr2, chr3)</span>
<span class="fu"><a href="../reference/get_input_segmentation.html">get_input_segmentation</a></span><span class="op">(</span><span class="va">congas_example</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">'chr'</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 19 × 12</span></span>
<span class="co">#&gt;    chr       start     end ploidy_real   tot   from     to   dist    mu fixed_mu</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1          1  1.26<span style="color: #949494;">e</span>7           1     1 1   <span style="color: #949494;">e</span>0 1.26<span style="color: #949494;">e</span>7 1.26<span style="color: #949494;">e</span>7    16       73</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   12<span style="text-decoration: underline;">600</span>001  6.02<span style="color: #949494;">e</span>7           2     2 1.26<span style="color: #949494;">e</span>7 6.02<span style="color: #949494;">e</span>7 4.75<span style="color: #949494;">e</span>7   102      302</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   61<span style="text-decoration: underline;">200</span>001  7.71<span style="color: #949494;">e</span>7           2     2 6.12<span style="color: #949494;">e</span>7 7.71<span style="color: #949494;">e</span>7 1.59<span style="color: #949494;">e</span>7    11       32</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   78<span style="text-decoration: underline;">750</span>001  1.56<span style="color: #949494;">e</span>8           2     2 7.88<span style="color: #949494;">e</span>7 1.56<span style="color: #949494;">e</span>8 7.72<span style="color: #949494;">e</span>7    87      261</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  156<span style="text-decoration: underline;">000</span>001  2.49<span style="color: #949494;">e</span>8           3     3 1.56<span style="color: #949494;">e</span>8 2.49<span style="color: #949494;">e</span>8 9.33<span style="color: #949494;">e</span>7   116      350</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr2          1  2.62<span style="color: #949494;">e</span>7           1     1 1   <span style="color: #949494;">e</span>0 2.62<span style="color: #949494;">e</span>7 2.62<span style="color: #949494;">e</span>7    15       56</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2   26<span style="text-decoration: underline;">250</span>001  3.81<span style="color: #949494;">e</span>7           2     2 2.63<span style="color: #949494;">e</span>7 3.81<span style="color: #949494;">e</span>7 1.18<span style="color: #949494;">e</span>7    19       59</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2   64<span style="text-decoration: underline;">050</span>001  8.72<span style="color: #949494;">e</span>7           2     2 6.41<span style="color: #949494;">e</span>7 8.72<span style="color: #949494;">e</span>7 2.31<span style="color: #949494;">e</span>7    35       95</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2   88<span style="text-decoration: underline;">350</span>001  1.60<span style="color: #949494;">e</span>8           2     2 8.84<span style="color: #949494;">e</span>7 1.60<span style="color: #949494;">e</span>8 7.15<span style="color: #949494;">e</span>7    48      159</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr2  159<span style="text-decoration: underline;">900</span>001  1.77<span style="color: #949494;">e</span>8           1     1 1.60<span style="color: #949494;">e</span>8 1.77<span style="color: #949494;">e</span>8 1.69<span style="color: #949494;">e</span>7    11       35</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr2  176<span style="text-decoration: underline;">850</span>001  1.95<span style="color: #949494;">e</span>8           2     2 1.77<span style="color: #949494;">e</span>8 1.95<span style="color: #949494;">e</span>8 1.86<span style="color: #949494;">e</span>7    12       39</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr2  201<span style="text-decoration: underline;">750</span>001  2.43<span style="color: #949494;">e</span>8           2     2 2.02<span style="color: #949494;">e</span>8 2.43<span style="color: #949494;">e</span>8 4.15<span style="color: #949494;">e</span>7    42      134</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr3          1  1.38<span style="color: #949494;">e</span>7           2     2 1   <span style="color: #949494;">e</span>0 1.38<span style="color: #949494;">e</span>7 1.38<span style="color: #949494;">e</span>7    12       41</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr3   34<span style="text-decoration: underline;">350</span>001  8.68<span style="color: #949494;">e</span>7           1     1 3.44<span style="color: #949494;">e</span>7 8.68<span style="color: #949494;">e</span>7 5.25<span style="color: #949494;">e</span>7    41      165</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">15</span> chr3   89<span style="text-decoration: underline;">100</span>001  1.14<span style="color: #949494;">e</span>8           2     2 8.91<span style="color: #949494;">e</span>7 1.14<span style="color: #949494;">e</span>8 2.49<span style="color: #949494;">e</span>7    15       46</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">16</span> chr3  114<span style="text-decoration: underline;">000</span>001  1.34<span style="color: #949494;">e</span>8           3     3 1.14<span style="color: #949494;">e</span>8 1.34<span style="color: #949494;">e</span>8 2.01<span style="color: #949494;">e</span>7    28       76</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">17</span> chr3  134<span style="text-decoration: underline;">700</span>001  1.52<span style="color: #949494;">e</span>8           3     3 1.35<span style="color: #949494;">e</span>8 1.52<span style="color: #949494;">e</span>8 1.74<span style="color: #949494;">e</span>7    22       55</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">18</span> chr3  152<span style="text-decoration: underline;">100</span>001  1.80<span style="color: #949494;">e</span>8           4     4 1.52<span style="color: #949494;">e</span>8 1.80<span style="color: #949494;">e</span>8 2.83<span style="color: #949494;">e</span>7    18       50</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">19</span> chr3  180<span style="text-decoration: underline;">450</span>001  1.98<span style="color: #949494;">e</span>8           1     1 1.80<span style="color: #949494;">e</span>8 1.98<span style="color: #949494;">e</span>8 1.77<span style="color: #949494;">e</span>7    17       63</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 2 more variables: segment_id &lt;chr&gt;, size &lt;int&gt;</span></span></code></pre></div>
<p>If you have run the main <code>rcongas</code> fit function
<code>XXX</code> by using parameters <code>XXXX</code>, the raw input
RNA data matrix have also been stored into the fit object (otherwise no,
just to save space in the final object).</p>
<p>You have a getter <code>get_input_raw_data</code> to obtain the input
data when this has been stored in a <code>rcongas</code> object, or you
need to load yourself the input data that you used to fit the model (in
that case <code>get_input_raw_data</code> would return <code>NULL</code>
and report a warning to screen). The raw input data are the counts, per
gene (rows), per cell (columns), obtained from the scRNAseq
experiment.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Input data - stored inside "congas_example" in this case</span>
<span class="va">raw_input_gene_by_cell</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_input_raw_data.html">get_input_raw_data</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>

<span class="co"># Just the 2x2 submatrix (genes on the rows, cells on the columns)</span>
<span class="co"># print(raw_input_gene_by_cell[1:2, 1:2])</span></code></pre></div>
<p>The data processed by CONGAS are counts mapped to segments, which are
stored in a matrix that contains cells on rows, and segments on columns.
This matrix can be extracted in long tidy format; the actual counts
value can be obtained before or after normalisation, and can be subset
to certain portions of the cancer genome.</p>
<p>In this long view we report also the cluster id for each of the input
cells.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Normalised data (default)</span>
<span class="fu"><a href="../reference/get_counts.html">get_counts</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 35,210 × 6</span></span>
<span class="co">#&gt;    chr    from       to cell                   n cluster</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAACCTGTCGACAGCC-1  22.0 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAAGATGGTAATAGCA-1  24.5 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACACGTAGCTATGCT-1  30.0 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTCCCAGACAGGCT-1  24.2 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTCCCGTACACCGC-1  30.6 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTGGTCAAAGAATC-1  18.6 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTGGTCAAAGTGCG-1  27.1 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTTTCGTGGCTCCA-1  29.1 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTTTCTCGGTTAAC-1  35.1 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAGACCTAGCCAGTAG-1  14.5 c2     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 35,200 more rows</span></span>

<span class="co"># A view on a certain chromosome </span>
<span class="fu"><a href="../reference/get_counts.html">get_counts</a></span><span class="op">(</span><span class="va">congas_example</span>,
           chromosomes <span class="op">=</span> <span class="st">'chr3'</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,521 × 6</span></span>
<span class="co">#&gt;    chr    from       to cell                   n cluster</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AAACCTGTCGACAGCC-1 24.6  c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AAAGATGGTAATAGCA-1 28.4  c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACACGTAGCTATGCT-1 28.4  c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTCCCAGACAGGCT-1 14.5  c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTCCCGTACACCGC-1 29.0  c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTGGTCAAAGAATC-1 16.4  c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTGGTCAAAGTGCG-1  5.41 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTTTCGTGGCTCCA-1 25.5  c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AACTTTCTCGGTTAAC-1 23.0  c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr3      1 13<span style="text-decoration: underline;">800</span>000 AAGACCTAGCCAGTAG-1 17.1  c2     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 3,511 more rows</span></span>

<span class="co"># Un-normalised data</span>
<span class="fu"><a href="../reference/get_counts.html">get_counts</a></span><span class="op">(</span><span class="va">congas_example</span>, normalise <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 35,210 × 6</span></span>
<span class="co">#&gt;    chr    from       to cell                   n cluster</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAACCTGTCGACAGCC-1    17 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAAGATGGTAATAGCA-1    19 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACACGTAGCTATGCT-1    18 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTCCCAGACAGGCT-1    20 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTCCCGTACACCGC-1    19 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTGGTCAAAGAATC-1    17 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTGGTCAAAGTGCG-1    15 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTTTCGTGGCTCCA-1    24 c1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AACTTTCTCGGTTAAC-1    32 c2     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1      1 12<span style="text-decoration: underline;">600</span>000 AAGACCTAGCCAGTAG-1    11 c2     </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 35,200 more rows</span></span></code></pre></div>
<p>If you want to know what genes have been used by CONGAS and to which
segments these have been mapped you can use function</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># All mapped genes</span>
<span class="fu"><a href="../reference/get_mapped_genes.html">get_mapped_genes</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 8,564 × 5</span></span>
<span class="co">#&gt;    gene          chr   from  to       segment_id     </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> AP006222.2    chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> RP11.206L10.9 chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> FAM41C        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> NOC2L         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ISG15         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> AGRN          chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> C1orf159      chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TTLL10        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PUSL1         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CPSF3L        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 8,554 more rows</span></span>

<span class="co"># Genes mapped in chromosomes '1' and '5'</span>
<span class="fu"><a href="../reference/get_mapped_genes.html">get_mapped_genes</a></span><span class="op">(</span><span class="va">congas_example</span>, chromosomes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'chr1'</span>, <span class="st">'chr5'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 1,377 × 5</span></span>
<span class="co">#&gt;    gene          chr   from  to       segment_id     </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> AP006222.2    chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> RP11.206L10.9 chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> FAM41C        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> NOC2L         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ISG15         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> AGRN          chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> C1orf159      chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> TTLL10        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PUSL1         chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> CPSF3L        chr1  1     12600000 chr1:1:12600000</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 1,367 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="post-clustering-analysis">Post-clustering analysis<a class="anchor" aria-label="anchor" href="#post-clustering-analysis"></a>
</h3>
<ul>
<li>
<code>get_DE_table</code> to obtain results from DE analysis;</li>
</ul>
<p>Results from Differential Expression (DE) analysis using
<code>Seurat</code> are available by using <code>get_DE_table</code>. By
default there are returned sorted by adjusted p-value, and filters can
be applied to subset by maximum adjusted p-value
(<code>cut_pvalue</code>), absolute log-fold change
(<code>cut_lfc</code>), and chromosome id.</p>
<p>Note that the filters are applied jointly (i.e., as
intersection).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default results (sorted by adjusted p-value, default filters)</span>
<span class="fu"><a href="../reference/get_DE_table.html">get_DE_table</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 162 × 8</span></span>
<span class="co">#&gt;    chr        from        to gene   pct_detec_1 ptc_detec_2 p_val_adj avg_log2FC</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chrX   71<span style="text-decoration: underline;">475</span>529  71<span style="text-decoration: underline;">497</span>150 RPS4X        1           1      2.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-82</span>     -<span style="color: #BB0000;">0.570</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chrX    1<span style="text-decoration: underline;">505</span>045   1<span style="text-decoration: underline;">511</span>617 SLC25…       0.997       1      1.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-81</span>     -<span style="color: #BB0000;">0.823</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chrX   73<span style="text-decoration: underline;">040</span>486  73<span style="text-decoration: underline;">072</span>588 XIST         0.045       0.879  5.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-54</span>     -<span style="color: #BB0000;">2.10</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr15  66<span style="text-decoration: underline;">790</span>355  66<span style="text-decoration: underline;">816</span>870 RPL4         1           1      8.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-44</span>      0.607</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr16   2<span style="text-decoration: underline;">012</span>053   2<span style="text-decoration: underline;">014</span>861 RPS2         1           1      9.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-36</span>      0.757</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr14 102<span style="text-decoration: underline;">547</span>075 102<span style="text-decoration: underline;">606</span>036 HSP90…       1           1      1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-35</span>      0.952</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1   39<span style="text-decoration: underline;">491</span>990  39<span style="text-decoration: underline;">500</span>308 NDUFS5       0.937       0.992  1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span>     -<span style="color: #BB0000;">0.734</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr6  133<span style="text-decoration: underline;">135</span>580 133<span style="text-decoration: underline;">138</span>703 RPS12        1           1      7.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-30</span>      0.363</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr18  43<span style="text-decoration: underline;">664</span>110  43<span style="text-decoration: underline;">684</span>300 ATP5A1       0.995       0.927  1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-28</span>      0.948</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr6   29<span style="text-decoration: underline;">909</span>037  29<span style="text-decoration: underline;">913</span>661 HLA-A        1           1      2.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-23</span>     -<span style="color: #BB0000;">0.869</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 152 more rows</span></span>

<span class="co"># DE filtered by p-value cut at alpha-level 5%</span>
<span class="fu"><a href="../reference/get_DE_table.html">get_DE_table</a></span><span class="op">(</span><span class="va">congas_example</span>, cut_pvalue <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 257 × 8</span></span>
<span class="co">#&gt;    chr        from        to gene   pct_detec_1 ptc_detec_2 p_val_adj avg_log2FC</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chrX   71<span style="text-decoration: underline;">475</span>529  71<span style="text-decoration: underline;">497</span>150 RPS4X        1           1      2.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-82</span>     -<span style="color: #BB0000;">0.570</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chrX    1<span style="text-decoration: underline;">505</span>045   1<span style="text-decoration: underline;">511</span>617 SLC25…       0.997       1      1.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-81</span>     -<span style="color: #BB0000;">0.823</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chrX   73<span style="text-decoration: underline;">040</span>486  73<span style="text-decoration: underline;">072</span>588 XIST         0.045       0.879  5.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-54</span>     -<span style="color: #BB0000;">2.10</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr15  66<span style="text-decoration: underline;">790</span>355  66<span style="text-decoration: underline;">816</span>870 RPL4         1           1      8.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-44</span>      0.607</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr16   2<span style="text-decoration: underline;">012</span>053   2<span style="text-decoration: underline;">014</span>861 RPS2         1           1      9.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-36</span>      0.757</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr14 102<span style="text-decoration: underline;">547</span>075 102<span style="text-decoration: underline;">606</span>036 HSP90…       1           1      1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-35</span>      0.952</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1   39<span style="text-decoration: underline;">491</span>990  39<span style="text-decoration: underline;">500</span>308 NDUFS5       0.937       0.992  1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span>     -<span style="color: #BB0000;">0.734</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr6  133<span style="text-decoration: underline;">135</span>580 133<span style="text-decoration: underline;">138</span>703 RPS12        1           1      7.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-30</span>      0.363</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr18  43<span style="text-decoration: underline;">664</span>110  43<span style="text-decoration: underline;">684</span>300 ATP5A1       0.995       0.927  1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-28</span>      0.948</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr6   29<span style="text-decoration: underline;">909</span>037  29<span style="text-decoration: underline;">913</span>661 HLA-A        1           1      2.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-23</span>     -<span style="color: #BB0000;">0.869</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 247 more rows</span></span>

<span class="co"># DE filtered to retain all values, regardless the LFC</span>
<span class="fu"><a href="../reference/get_DE_table.html">get_DE_table</a></span><span class="op">(</span><span class="va">congas_example</span>, cut_lfc <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 165 × 8</span></span>
<span class="co">#&gt;    chr        from        to gene   pct_detec_1 ptc_detec_2 p_val_adj avg_log2FC</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chrX   71<span style="text-decoration: underline;">475</span>529  71<span style="text-decoration: underline;">497</span>150 RPS4X        1           1      2.34<span style="color: #949494;">e</span><span style="color: #BB0000;">-82</span>     -<span style="color: #BB0000;">0.570</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chrX    1<span style="text-decoration: underline;">505</span>045   1<span style="text-decoration: underline;">511</span>617 SLC25…       0.997       1      1.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-81</span>     -<span style="color: #BB0000;">0.823</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chrX   73<span style="text-decoration: underline;">040</span>486  73<span style="text-decoration: underline;">072</span>588 XIST         0.045       0.879  5.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-54</span>     -<span style="color: #BB0000;">2.10</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr15  66<span style="text-decoration: underline;">790</span>355  66<span style="text-decoration: underline;">816</span>870 RPL4         1           1      8.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-44</span>      0.607</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr16   2<span style="text-decoration: underline;">012</span>053   2<span style="text-decoration: underline;">014</span>861 RPS2         1           1      9.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-36</span>      0.757</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr14 102<span style="text-decoration: underline;">547</span>075 102<span style="text-decoration: underline;">606</span>036 HSP90…       1           1      1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-35</span>      0.952</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1   39<span style="text-decoration: underline;">491</span>990  39<span style="text-decoration: underline;">500</span>308 NDUFS5       0.937       0.992  1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span>     -<span style="color: #BB0000;">0.734</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr6  133<span style="text-decoration: underline;">135</span>580 133<span style="text-decoration: underline;">138</span>703 RPS12        1           1      7.85<span style="color: #949494;">e</span><span style="color: #BB0000;">-30</span>      0.363</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr18  43<span style="text-decoration: underline;">664</span>110  43<span style="text-decoration: underline;">684</span>300 ATP5A1       0.995       0.927  1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-28</span>      0.948</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr6   29<span style="text-decoration: underline;">909</span>037  29<span style="text-decoration: underline;">913</span>661 HLA-A        1           1      2.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-23</span>     -<span style="color: #BB0000;">0.869</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 155 more rows</span></span>

<span class="co"># A view on a certain chromosome </span>
<span class="fu"><a href="../reference/get_DE_table.html">get_DE_table</a></span><span class="op">(</span><span class="va">congas_example</span>, chromosomes <span class="op">=</span> <span class="st">'chr15'</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 8</span></span>
<span class="co">#&gt;    chr        from        to gene   pct_detec_1 ptc_detec_2 p_val_adj avg_log2FC</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr15  66<span style="text-decoration: underline;">790</span>355  66<span style="text-decoration: underline;">816</span>870 RPL4         1           1      8.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-44</span>      0.607</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr15  45<span style="text-decoration: underline;">003</span>675  45<span style="text-decoration: underline;">011</span>075 B2M          1           1      2.01<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span>     -<span style="color: #BB0000;">0.722</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr15  79<span style="text-decoration: underline;">102</span>829  79<span style="text-decoration: underline;">190</span>475 MORF4…       0.984       0.952  2.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span>      0.796</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr15  78<span style="text-decoration: underline;">832</span>747  78<span style="text-decoration: underline;">841</span>604 PSMA4        0.997       0.935  6.35<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span>      0.805</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr15 101<span style="text-decoration: underline;">821</span>715 101<span style="text-decoration: underline;">835</span>487 SNRPA1       0.939       0.79   1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span>      0.804</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr15  85<span style="text-decoration: underline;">212</span>775  85<span style="text-decoration: underline;">259</span>947 SEC11A       0.966       0.79   5.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-12</span>      0.696</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr15  78<span style="text-decoration: underline;">632</span>666  78<span style="text-decoration: underline;">640</span>572 CRABP1       0.782       0.419  1.98<span style="color: #949494;">e</span><span style="color: #BB0000;">-11</span>      0.888</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr15  65<span style="text-decoration: underline;">822</span>756  65<span style="text-decoration: underline;">870</span>687 PTPLA…       0.974       0.895  5.29<span style="color: #949494;">e</span><span style="color: #BB0000;">- 9</span>      0.679</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr15  75<span style="text-decoration: underline;">212</span>132  75<span style="text-decoration: underline;">230</span>509 COX5A        0.855       0.548  2.33<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      0.604</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr15  90<span style="text-decoration: underline;">626</span>277  90<span style="text-decoration: underline;">645</span>736 IDH2         0.747       0.452  3.10<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      0.659</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr15  81<span style="text-decoration: underline;">239</span>667  81<span style="text-decoration: underline;">282</span>219 MESDC2       0.695       0.339  3.98<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      0.609</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr15  78<span style="text-decoration: underline;">570</span>177  78<span style="text-decoration: underline;">592</span>136 WDR61        0.734       0.452  9.71<span style="color: #949494;">e</span><span style="color: #BB0000;">- 6</span>      0.596</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr15  64<span style="text-decoration: underline;">979</span>772  64<span style="text-decoration: underline;">995</span>480 OAZ2         0.892       0.71   1.65<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>      0.538</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr15  64<span style="text-decoration: underline;">448</span>011  64<span style="text-decoration: underline;">455</span>404 PPIB         0.984       0.935  2.20<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>      0.575</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">15</span> chr15  64<span style="text-decoration: underline;">657</span>193  64<span style="text-decoration: underline;">679</span>886 KIAA0…       0.726       0.452  5.60<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span>      1.09</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://sites.google.com/view/cdslab" class="external-link">Salvatore
Milite</a>, <a href="https://sites.google.com/view/cdslab" class="external-link">Giulio
Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
