<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Plotting functions • Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Plotting functions">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a></a>
    </li>
    <li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/3_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/4_Preprocessing.html">Preprocessing</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Old version</li>
    <li>
      <a href="../old_vignettes/kieran_vignette.html">Breast PDX analysis</a>
    </li>
    <li>
      <a href="../old_vignettes/Glioblastoma_smart_seq_analysis.html">Glioblastoma GBM analysis</a>
    </li>
    <li>
      <a href="../old_vignettes/Glioblastoma_smart_seq_analysis.html">Glioblastoma GBM analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/militeee/Rcongas/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="3_Plotting_functions_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Plotting functions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/militeee/Rcongasvignettes/3_Plotting_functions.Rmd"><code>vignettes/3_Plotting_functions.Rmd</code></a></small>
      <div class="hidden name"><code>3_Plotting_functions.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span><span class="op">(</span><span class="va">Rcongas</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: Rcongas</span>
<span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span>
<span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'BMix'</span>
<span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span>
<span class="co">#&gt; 'BMix'</span>
<span class="co">#&gt; ✓ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'tibble' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'tidyr' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'readr' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'stringr' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'forcats' was built under R version 4.1.1</span>
<span class="co">#&gt; Warning: package 'crayon' was built under R version 4.1.1</span>
<span class="co">#&gt; ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</span>
<span class="co">#&gt; ✓ Loading Rcongas, 'Copy-Number genotyping from single cells'. Support : &lt;https://militeee.github.io/Rcongas/&gt;</span></code></pre></div>
<p>We show how you can plot information from a CONGAS object. We work with an example object released with the tool.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">'congas_example'</span>, package <span class="op">=</span> <span class="st">'Rcongas'</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; ── [ Rcongas ]  ────────────────────────────────────────────────────────────────</span>
<span class="co">#&gt; → Data: 503 cells with 8564 genes, aggregated in 70 segments.</span>
<span class="co">#&gt; ℹ Clusters: k = 2, model with AIC = 577427.35.</span>
<span class="co">#&gt; ● Cluster c1, n = 379 [75.35% of total cells].</span>
<span class="co">#&gt; ● Cluster c2, n = 124 [24.65% of total cells].</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── CNA highlights (alpha = 0.05)</span>
<span class="co">#&gt; ✓  c1 CNA(s):  [chr14:93300001:105450000] vs c2, [chr15:67050001:102600000] vs c2, [chr16:1:3750000] vs c2, and [chr18:32400001:55950000] vs c2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Differential Expression DESeq2  (1 vs 2)</span>
<span class="co">#&gt; ✓ 212 DE genes (alpha = 0.01, |lfc| &gt; 0.25).</span></code></pre></div>
<div id="genome-wide-clone-profiles" class="section level2">
<h2 class="hasAnchor">
<a href="#genome-wide-clone-profiles" class="anchor"></a>Genome-wide clone profiles</h2>
<p>Function <code>plot_gw_cna_profiles</code> allows to plot genome-wide clone-specific CN profiles inferred by CONGAS.</p>
<p>Since segments are independent in the underlying statistical model and modelled by using lognormal distributions, each segment is scaled to reflect the offset to the mean estimated from all che model clusters. This type of measurement is called Normalised Copy Number.</p>
<p>Note that this is the only way of making sense in the comparison among the CN values obtained for different segments. It is not trivial to determine how different two segmetns should be to reflect an overall change in ploidy in the clone; nonetheless, this kind of scaling allows to detect which segments show the largest CN difference among the identified clones.</p>
<p>The default view splits works by facetting the chromosomes. The gray bar in eah chromosome represents the centromeres start/ end, determined from the used reference genome.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default, with facets</span>
<span class="fu"><a href="../reference/plot_gw_cna_profiles.html">plot_gw_cna_profiles</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-3-1.png" width="768"></p>
<p>An alternative view is available to pool all the chromosomes together</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A whole-genome view</span>
<span class="fu"><a href="../reference/plot_gw_cna_profiles.html">plot_gw_cna_profiles</a></span><span class="op">(</span><span class="va">congas_example</span>, whole_genome <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-4-1.png" width="768"></p>
<p>It is possible to zoom in chromosomes with this plot (the dark gray bar is the centromere).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># A whole-genome view for chromosome 4</span>
<span class="fu"><a href="../reference/plot_gw_cna_profiles.html">plot_gw_cna_profiles</a></span><span class="op">(</span><span class="va">congas_example</span>, whole_genome <span class="op">=</span> <span class="cn">TRUE</span>, chromosomes <span class="op">=</span> <span class="st">'chr4'</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-5-1.png" width="384"></p>
</div>
<div id="chromosomes-highlight" class="section level2">
<h2 class="hasAnchor">
<a href="#chromosomes-highlight" class="anchor"></a>Chromosomes highlight</h2>
<p>The criterion to highlight a CNA in a fit is determined from this distribution with a parameter <code>alpha</code> that sets the vertical dashed line (is a quantile).</p>
<p>This is a heuristic.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default</span>
<span class="fu"><a href="../reference/plot_highlights.html">plot_highlights</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># More stringent</span>
<span class="fu"><a href="../reference/plot_highlights.html">plot_highlights</a></span><span class="op">(</span><span class="va">congas_example</span>, alpha <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-6-2.png" width="768"></p>
</div>
<div id="plotting-specific-chromosomes" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-specific-chromosomes" class="anchor"></a>Plotting specific chromosomes</h2>
<p>You can plot a chromosome by specifying its id. The height of the plot should be adjusted depending on the number of segments in the chromosome. If DE analysis is available (see below), a chromosome specific volcano plot is added</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot chromosome 1 (labeled "chr1")</span>
<span class="fu"><a href="../reference/plot_chromosome.html">plot_chromosome</a></span><span class="op">(</span><span class="va">congas_example</span>, chr <span class="op">=</span> <span class="st">'chr1'</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-7-1.png" width="672"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Plot chromosome 15 (labeled "chr15")</span>
<span class="fu"><a href="../reference/plot_chromosome.html">plot_chromosome</a></span><span class="op">(</span><span class="va">congas_example</span>, chr <span class="op">=</span> <span class="st">'chr15'</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-8-1.png" width="672"></p>
</div>
<div id="plotting-input-segments-data" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-input-segments-data" class="anchor"></a>Plotting input segments data</h2>
<p>You can visualise the actual input data used by CONGAS, using function <code>plot_counts_rna_segments</code>: these are the counts per segments, after genes have been mapped and filtered as explained in <code>XXXX</code>.</p>
<p>The function plots a heatmap-style view of the data, with associated the clustering assignments of each cell (rows). In order to plot these data you need to retr</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Use z-score, using counts normalisation</span>
<span class="fu"><a href="../reference/plot_counts_rna_segments.html">plot_counts_rna_segments</a></span><span class="op">(</span><span class="va">congas_example</span>, z_score <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-9-1.png" width="480"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Without counts normalisation and without z-score.</span>
<span class="fu"><a href="../reference/plot_counts_rna_segments.html">plot_counts_rna_segments</a></span><span class="op">(</span><span class="va">congas_example</span>, normalised <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-9-2.png" width="480"></p>
</div>
<div id="plotting-input-raw-data" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-input-raw-data" class="anchor"></a>Plotting input raw data</h2>
<p>Raw data in input to CONGAS are counts of transcripts per cell. This matrix can be extremely large, and it should be plotted only on a subset of the data in many cases.</p>
<p>This function takes as input a list of genes to subset the plot. Also, by default prompts the user to assess if he/she really wants to plot the data; this helps because sometimes if the input data are too large, the plot becomes useless (and can freeze the computation). If you seek to skip the prompr you should set <code>prompt = FALSE</code>.</p>
<p>The plot can be generated with, and without clustering assignments.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Raw input counts. Since we have them stored in "congas_example",</span>
<span class="co"># we can access them straight from the object</span>
<span class="va">raw_input</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_input_raw_data.html">get_input_raw_data</a></span><span class="op">(</span><span class="va">congas_example</span>, transpose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>

<span class="co"># Used genes - just the a random set of 30. Wem use the "get_mapped_genes"</span>
<span class="co"># function to get the ones actually used in "congas_example"</span>
<span class="va">random_30_genes</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_mapped_genes.html">get_mapped_genes</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/sample_n.html">sample_n</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span>

<span class="co"># Plot</span>
<span class="fu"><a href="../reference/plot_raw_data.html">plot_raw_data</a></span><span class="op">(</span>
  x <span class="op">=</span> <span class="va">raw_input</span>,
  lognormalise <span class="op">=</span> <span class="cn">T</span>,
  genes <span class="op">=</span> <span class="va">random_30_genes</span>,
  prompt <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Plot with clusters</span>
<span class="fu"><a href="../reference/plot_raw_data.html">plot_raw_data</a></span><span class="op">(</span>
  x <span class="op">=</span> <span class="va">raw_input</span>,
  genes <span class="op">=</span> <span class="va">random_30_genes</span>,lognormalise <span class="op">=</span> <span class="cn">T</span>,
  clusters <span class="op">=</span> <span class="fu"><a href="../reference/get_clusters.html">get_clusters</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>,
  prompt <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># Now we use the DE genes (see below)</span>
<span class="fu"><a href="../reference/plot_raw_data.html">plot_raw_data</a></span><span class="op">(</span>
  x <span class="op">=</span> <span class="va">raw_input</span>,
  genes <span class="op">=</span> <span class="fu"><a href="../reference/get_DE_table.html">get_DE_table</a></span><span class="op">(</span> <span class="co"># DE genes selected</span>
    <span class="va">congas_example</span>, 
    cut_pvalue <span class="op">=</span> <span class="fl">0.001</span>, 
    cut_lfc <span class="op">=</span> <span class="fl">.75</span><span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html">pull</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span>,
  clusters <span class="op">=</span> <span class="fu"><a href="../reference/get_clusters.html">get_clusters</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>,
  lognormalise <span class="op">=</span> <span class="cn">T</span>,
  prompt <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
</div>
<div id="plotting-de-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-de-analysis" class="anchor"></a>Plotting DE analysis</h2>
<p>The most common plot that we implemented for a Differential Expression analysis is a volcano plot, reporting on the x-axis the LFC (log-fold change), and on the y-axis the test p-value (adjusted).</p>
<p>The function has two cutoffs options to highlight significantly DEGs (Differentially Expressed Genes), and annotate the top ones (both Upregulated, with <code>LFC &gt; 0</code>, and Downregulated, with <code>LFC &lt; 0</code>).</p>
<p>A gene is deemed non-significant <code>ns</code> based on a joint condition that involved LFC and the p-value.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot</span>
<span class="fu"><a href="../reference/plot_DE_volcano.html">plot_DE_volcano</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-11-1.png" width="480"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Setting the cut on LFC to 0</span>
<span class="fu"><a href="../reference/plot_DE_volcano.html">plot_DE_volcano</a></span><span class="op">(</span><span class="va">congas_example</span>, cut_pvalue <span class="op">=</span> <span class="fl">0.05</span>, cut_lfc <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-11-2.png" width="480"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Requiring lower p-value, decreasing LFC, and annotating top 10 genes</span>
<span class="fu"><a href="../reference/plot_DE_volcano.html">plot_DE_volcano</a></span><span class="op">(</span>
  <span class="va">congas_example</span>,
  annotate_top <span class="op">=</span> <span class="fl">10</span>,
  cut_pvalue <span class="op">=</span> <span class="fl">1e-6</span>,
  cut_lfc <span class="op">=</span> <span class="fl">0.15</span>
<span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-11-3.png" width="480"></p>
<p>It is possible to integrate the whole-genome view of the copy number segments, together with the results from a DE analysis.</p>
<p>The combined plot shows, on top of the segments, the total number of deferentially expressed genes (DEGs) that map to the segments as well as their relative frequency with respect to genes that map to the segments, and that have beend tested for DE status. On top, the plot reports also a line plot showing where other DEGs have been found, which however map off the CNA segments used by CONGAS.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Default plot</span>
<span class="fu"><a href="../reference/plot_DE_gw.html">plot_DE_gw</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_rect).</span>

<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
<p>The counts distribution per segment, with annotated quantiles.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/plot_counts_per_segment.html">plot_counts_per_segment</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-13-1.png" width="1152"></p>
<p>The cohort statistics (three plots):</p>
<ul>
<li>percentage of 0s in cell counts (to find cells with little count information)</li>
<li>counts distribution across all cohort (to find outliers in count data)</li>
<li>percentage of 0s in gene counts (to find genes expressed in a little portion of the population)</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Returns a number of plots with different statistics</span>
<span class="fu"><a href="../reference/plot_cohort_statistics.html">plot_cohort_statistics</a></span><span class="op">(</span><span class="va">congas_example</span>, assembly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; $zeroes</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-14-1.png" width="384"></p>
<pre><code>#&gt; 
#&gt; $counts</code></pre>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-14-2.png" width="384"></p>
<pre><code>#&gt; 
#&gt; $genes</code></pre>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-14-3.png" width="384"></p>
</div>
<div id="analysis-reports" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-reports" class="anchor"></a>Analysis report(s)</h2>
<p>A full report assembles some of these visualisations.</p>
<p>This is a standard analysis report.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/report_analysis.html">report_analysis</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; ✓ Loading ctree, 'Clone trees in cancer'. Support : &lt;https://caravagn.github.io/ctree/&gt;</span>
<span class="co">#&gt; Warning: replacing previous import 'Biostrings::translate' by</span>
<span class="co">#&gt; 'seqinr::translate' when loading 'dndscv'</span>
<span class="co">#&gt; ✔ Loading mobster, 'Model-based clustering in cancer'. Support :</span>
<span class="co">#&gt; &lt;https://caravagn.github.io/mobster/&gt;.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>

<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>

<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-15-1.png" width="1536"></p>
<p>This is to focus on Differential Expressions.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/report_analysis_de.html">report_analysis_de</a></span><span class="op">(</span><span class="va">congas_example</span><span class="op">)</span>
<span class="co">#&gt; Warning: `guides(&lt;scale&gt; = FALSE)` is deprecated. Please use `guides(&lt;scale&gt; =</span>
<span class="co">#&gt; "none")` instead.</span>
<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_rect).</span>

<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_rect).</span>
<span class="co">#&gt; Warning: Removed 16 rows containing missing values (geom_point).</span></code></pre></div>
<p><img src="3_Plotting_functions_files/figure-html/unnamed-chunk-16-1.png" width="1536"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/view/cdslab">Salvatore Milite</a>, <a href="https://sites.google.com/view/cdslab">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
