<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Preprocessing • Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Preprocessing">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a></a>
    </li>
    <li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/3_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/4_Preprocessing.html">Preprocessing</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Old version</li>
    <li>
      <a href="../../old_vignettes/kieran_vignette.html">Breast PDX analysis</a>
    </li>
    <li>
      <a href="../../old_vignettes/Glioblastoma_smart_seq_analysis.html">Glioblastoma GBM analysis</a>
    </li>
    <li>
      <a href="../../old_vignettes/monosomy_vignette.html">Monosomy 7 analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/militeee/Rcongas/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="4_Propreocessing_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Preprocessing</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/militeee/Rcongasvignettes/4_Propreocessing.Rmd"><code>vignettes/4_Propreocessing.Rmd</code></a></small>
      <div class="hidden name"><code>4_Propreocessing.Rmd</code></div>

    </div>

    
    
<p>As the joint calling of CN alteration, data normalization and clustering is extremely challenging and characterized by different degree of freedom a filtering step is essential to ensure we are not inflating our false positive count. First of all a filter on highly variable and highly expressed genes can be extremely useful as it tends to remove cell specific markers and those genes that can create significant deviations in the total counts over segment.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: package 'Seurat' was built under R version 4.1.1</code></pre>
<pre><code>## Attaching SeuratObject</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">Rcongas</span><span class="op">)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by 'cli':
##   method     from         
##   print.boxx spatstat.geom</code></pre>
<pre><code>## Warning: replacing previous import 'cli::num_ansi_colors' by
## 'crayon::num_ansi_colors' when loading 'BMix'</code></pre>
<pre><code>## Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading
## 'BMix'</code></pre>
<pre><code>## ✓ Loading BMix, 'Binomial and Beta-Binomial univariate mixtures'. Support : &lt;https://caravagnalab.github.io/BMix/&gt;</code></pre>
<pre><code>## Warning: package 'ggplot2' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'tibble' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'tidyr' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'readr' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'stringr' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'forcats' was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package 'crayon' was built under R version 4.1.1</code></pre>
<pre><code>## ✓ Loading CNAqc, 'Copy Number Alteration quality check'. Support : &lt;https://caravagn.github.io/CNAqc/&gt;</code></pre>
<pre><code>## ✓ Loading Rcongas, 'Copy-Number genotyping from single cells'. Support : &lt;https://militeee.github.io/Rcongas/&gt;</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>

<span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">Rcongas</span><span class="fu">::</span><span class="va"><a href="../reference/congas_example.html">congas_example</a></span>

<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_input_raw_data.html">get_input_raw_data</a></span><span class="op">(</span><span class="va">x</span>, as_matrix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">counts_stab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html">FindVariableFeatures</a></span><span class="op">(</span><span class="va">counts</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  <span class="va">t</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">counts_stab</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>  <span class="fu"><a href="https://rdrr.io/r/base/make.names.html">make.names</a></span><span class="op">(</span>unique <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>

<span class="co"># we just need a dataframe with genes as rownames and the mean and the variance respectively on</span>
<span class="co"># the first and second column </span>

<span class="va">to_remove</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_genes.html">filter_genes</a></span><span class="op">(</span><span class="va">counts_stab</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span>, plot <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="4_Propreocessing_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Now our count matrix is ready for the CONGAS analysis </span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">counts</span><span class="op">[</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">to_remove</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<p>Another difficult task is to choice of the right prior for the library size factors as it affects a lot the final quality of the inference. If you are not sure or to get a decent starting point Rcongas provides a function to fit an hypothetical library size distribution under tha monoclonal assumption, that can be a good starting point.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">Rcongas</span><span class="fu">::</span><span class="va"><a href="../reference/congas_example.html">congas_example</a></span>


<span class="va">coeff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/approx_theta_prior_params.html">approx_theta_prior_params</a></span><span class="op">(</span><span class="va">x</span>, plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="4_Propreocessing_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># The function provides a multiplicative factor for the shape and the rate, </span>
<span class="co"># this gives us a sort of control over the variance.</span>
<span class="co"># Remember that values less than one increase the variance, while values </span>
<span class="co"># values larger than one decrese the variance</span>
<span class="va">coeff_large</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/approx_theta_prior_params.html">approx_theta_prior_params</a></span><span class="op">(</span><span class="va">x</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, mult_factor <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="4_Propreocessing_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coeff_small</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/approx_theta_prior_params.html">approx_theta_prior_params</a></span><span class="op">(</span><span class="va">x</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, mult_factor <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="4_Propreocessing_files/figure-html/unnamed-chunk-2-3.png" width="700"></p>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/view/cdslab">Salvatore Milite</a>, <a href="https://sites.google.com/view/cdslab">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
