<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Rcongas ‚Ä¢ Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Rcongas">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Tutorial.html">Tutorial</a>
    </li>
    <li>
      <a href="../articles/3_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/4_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/5_Preprocessing.html">Preprocessing</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/militeee/Rcongas/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Rcongas</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/militeee/Rcongasvignettes/Rcongas.Rmd" class="external-link"><code>vignettes/Rcongas.Rmd</code></a></small>
      <div class="hidden name"><code>Rcongas.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Rcongas is a package to cluster cells according to their copy-number
profile and infer the corresponding CNV value w.r.t. a baseline bulk
sequencing reference.</p>
<p>The tool requires as input:</p>
<ul>
<li><p>A gene by cells count matrix with Hugo Symbols as row names and
cell names as column names</p></li>
<li><p>A copy number table obtained from bulk sequencing or direct
RNA-seq segmentation. The table needs to have to following mandatory
column: chr, from, to and tot (a numeric value indicating the bulk or
average copy number value for that segment)</p></li>
</ul>
<p>Rcongas performs a joint clustering and CNV value inference procedure
using a Poisson or Gaussian Mixture Model with prior distributions
informed by the copy number table obtained in input.</p>
<p>Rcongas can be used to genotype subclonal population characterized by
prototypical diversity.</p>
<div class="section level5">
<h5 id="the-model">The model<a class="anchor" aria-label="anchor" href="#the-model"></a>
</h5>
<p><span class="math inline">\(Y\)</span> is the <span class="math inline">\(ùëÅ √ó ùêº\)</span> input data matrix of RNA counts,
which describe <span class="math inline">\(ùëÅ\)</span> sequenced cells
and <span class="math inline">\(I\)</span> input segments (mapped
anywhere on the genome). Counts on a segment are summed up by pooling
all genes <span class="math inline">\(y_{n,i}\)</span> that map to the
segment. The final likelihood can be written as:</p>
<p><span class="math display">\[
p(\mathbf{Y} \mid \boldsymbol{\theta}, \boldsymbol{\mu}, \mathbf{C},
\mathbf{Z}, \boldsymbol{\pi})=\prod_{n=1}^{N} \prod_{i=1}^{I}
p\left(y_{n i} \mid \boldsymbol{\theta}, \boldsymbol{\mu}, \mathbf{C},
\mathbf{Z}, \boldsymbol{\pi}\right)
\]</span></p>
<p>And the segment likelihood is:</p>
<p><span class="math display">\[
p\left(y_{n i} \mid \boldsymbol{\theta}, \boldsymbol{\mu}, \mathbf{C},
\mathbf{Z}\right)=\operatorname{Pois}\left(\frac{\theta_{n} \cdot
\mu_{i} \cdot \prod_{k=1}^{K} C_{i k}^{z_{n k}}}{\sum_{i=1}^{I}
\prod_{k=1}^{K} C_{i k}^{z_{n k}}}\right)
\]</span></p>
<p>where the model uses a Gamma-distributed latent variable <span class="math inline">\(\theta_n\)</span> which models the library size
for cell <span class="math inline">\(n\)</span>, and <span class="math inline">\(\mu_i\)</span> for the number of genes in <span class="math inline">\(i\)</span> segment (a constant determined from
data). In CONGAS <span class="math inline">\(\mathbf{C}\)</span> is the
CNA profile for clones, where each clone <span class="math inline">\(k\)</span> is defined by segments and ùêº associated
CNAs; the prior for <span class="math inline">\(\mathbf{C}\)</span> is a
log-transform of a normal distribution, consistently with the fact that
ploidies are positive values. In this formulation <span class="math inline">\(\mathbf{Z}\)</span> are the <span class="math inline">\(N \times K\)</span> latent variables that assign
cells to clusters, and <span class="math inline">\(\pi\)</span> the
<span class="math inline">\(K\)</span>-dimensional mixing
proportions.</p>
<p>The model joint distribution can be factored as <span class="math display">\[
p(Y, Z, C, \theta, \pi)=p(Y \mid C, \theta, \pi) p(Z \mid \pi) p(\pi)
\prod_{i k} p\left(c_{i k}\right) \prod_{n} p\left(\theta_{n}\right)
\]</span> and in the variational framework latent variables are
approximated as variational distributions <span class="math inline">\(q(Z, C, \theta, \pi)\)</span>, supposed to be
independent and factorizable. The prior distributions for our latent
variables are:</p>
<ul>
<li><p><span class="math inline">\(p\left(c_{i k}\right) \sim \log
\operatorname{Norm}\left(m_{i k}, v\right)\)</span>, where <span class="math inline">\(m_{i k}\)</span> is the input CNA value from bulk
DNA, and the variance <span class="math inline">\(v\)</span> governs how
far the actual CNAs can be compared to input (default <span class="math inline">\(v=0.5\)</span> );</p></li>
<li><p><span class="math inline">\(p\left(\theta_{n}\right) \sim
\operatorname{Gamma}\left(e_{s^{\prime}} e_{r}\right)\)</span>, a
scarcely informative prior that works well in most cases (default <span class="math inline">\(e_{s}=3, e_{r}=1\)</span> );</p></li>
<li><p><span class="math inline">\(p(\pi) \sim
\operatorname{Dirichlet}(r)\)</span>, a prior over cluster
distributions, by default all assumed to have equal proportions
(i.e.¬†<span class="math inline">\(r=1 / k\)</span> ).</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="achtung">Achtung!<a class="anchor" aria-label="anchor" href="#achtung"></a>
</h4>
<p>The model has the following statistical assumptions:</p>
<ul>
<li><p>There is linear correlation between the number of copies of a
segment of DNA and the average expression of the genes mapping in that
segment.</p></li>
<li><p>Library size factors are assumed to be unimodal, if it is not the
case we suggest to split the dataset and run independent analysis for
each unimodal subgroup.</p></li>
<li><p>Copy number variation is the main biological driver of expression
variance in a segment when considering all the gene mapping on
it.</p></li>
</ul>
<p><em>They may not be valid in all the possible biological and
technical conditions and can be the main reason of false positive calls.
Most of them can be solved through pre-processing others require
specific attention.</em></p>
<p>We know this assumptions are super strong and we are actually working
on a new version of the tool the relaxes some constraints, provides
support for new datatypes and uses some better statistics. Endure with
us a little longer.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://sites.google.com/view/cdslab" class="external-link">Salvatore
Milite</a>, <a href="https://sites.google.com/view/cdslab" class="external-link">Giulio
Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
