<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Rcongas • Rcongas</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Rcongas">
<meta property="og:description" content="Rcongas">
<meta property="og:image" content="caravagnalab.github.io/Rcongas/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/1_create_congas_object.html">Object Initialization and preprocessing</a>
    </li>
    <li>
      <a href="../articles/2_Fitting.html">Fitting a model</a>
    </li>
    <li>
      <a href="../articles/map_genes_featues.html">Gene mapping</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagnalab/Rcongas/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Rcongas</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/Rcongasvignettes/Rcongas.Rmd" class="external-link"><code>vignettes/Rcongas.Rmd</code></a></small>
      <div class="hidden name"><code>Rcongas.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">Rcongas</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="analysis-workflow">Analysis workflow<a class="anchor" aria-label="anchor" href="#analysis-workflow"></a>
</h2>
<p>These data are required:</p>
<ul>
<li>single-cell RNA or ATAC sequencing data from independent cells, in
the form of counts per cell;</li>
<li>genome-wide segmentation with associated per-segment ploidy.</li>
</ul>
<p>The formats of the inputs is discussed here.</p>
<p>The (R)CONGAS+ workflow consists in:</p>
<ul>
<li>(optional) applying <a href="1_create_congas_object.html">filters</a> on the input data;</li>
<li>creating an (R)CONGAS+ object;</li>
<li>(optional) applying <a href="1_create_congas_object.html">filters</a> on the mapped data;</li>
<li>fitting a model</li>
</ul>
<p>The package comes with input data to show what the required input
formats are, and ready to use objects for case studies that we have
discussed in the main papers; see <a href="../docs/reference/index.html">References</a> for details.</p>
</div>
<div class="section level2">
<h2 id="input-formats">Input formats<a class="anchor" aria-label="anchor" href="#input-formats"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example data available in the package</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">'multiome_congas_object'</span><span class="op">)</span></span></code></pre></div>
<p>RNA and ATAC have the same input format reporting:</p>
<ul>
<li>
<code>chr</code>, <code>from</code> and <code>to</code>: genomic
locations of a gene (RNA) or a peak (ATAC)</li>
<li>
<code>cell</code>: a cell identifier</li>
<li>
<code>value</code>: the observed counts (discrete) This tibble is
created woth the function </li>
</ul>
<p>Segments have this format:</p>
<ul>
<li>
<code>chr</code>, <code>from</code> and <code>to</code>: genomic
range of the segment</li>
<li>
<code>copies</code>: ploidy of each segment</li>
</ul>
</div>
<div class="section level2">
<h2 id="creation-of-a-new-dataset">Creation of a new dataset<a class="anchor" aria-label="anchor" href="#creation-of-a-new-dataset"></a>
</h2>
<div class="section level3">
<h3 id="multiomodal-assay">multiomodal assay<a class="anchor" aria-label="anchor" href="#multiomodal-assay"></a>
</h3>
<p>Please see the <a href="1_create_congas_object.html">RCONGAS object
creation vignette</a> to understand how to create a RCONGAS+ object.
Once it is successfully generated, we can visualize some information
about the object</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiome_congas_object</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="va">print</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #BB0000; font-weight: bold;">Warning</span> <span style="color: #BB0000;">ATAC 0-counts cells.</span> <span style="color: #00BB00;">2</span> cells have no data in any of <span style="color: #00BB00;">22</span> segments, top <span style="color: #00BB00;">2</span> with missing data are:</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Cell <span style="color: #00BB00;">GTCCTAGAGCCAAATC-1-ATAC</span> with <span style="color: #BB0000;">1</span> 0-segments (<span style="color: #BB0000;">5%</span>)</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> Cell <span style="color: #00BB00;">TCTCGCCCAAACATAG-1-ATAC</span> with <span style="color: #BB0000;">1</span> 0-segments (<span style="color: #BB0000;">5%</span>)</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ (R)CONGAS+ ]</span><span style="color: #0000BB;"> Bimodal lymphoma</span> ─────────────────────────────────────────────</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── CNA segments (reference: <span style="color: #00BB00;">GRCh38</span>)</span></span>
<span><span class="co">#&gt; → Input <span style="color: #00BB00;">22</span> CNA segments, mean ploidy <span style="color: #00BB00;">2</span>.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  <span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;"> </span>|<span style="background-color: #00BB00;"> </span>|<span style="background-color: #00BB00;"> </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;"> </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span>|<span style="background-color: #00BB00;">  </span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   <span style="text-decoration: underline;">Ploidy:</span>  <span style="background-color: #00BBBB;"> </span> 0   <span style="background-color: #0000BB;"> </span> 1   <span style="background-color: #00BB00;"> </span> 2   <span style="background-color: #BBBB00;"> </span> 3   <span style="background-color: #BB0000;"> </span> 4   <span style="background-color: #BB00BB;"> </span> 5   <span style="background-color: #BBBBBB;"> </span> *</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Modalities</span></span>
<span><span class="co">#&gt; → RNA: <span style="color: #00BB00;">500</span> cells with <span style="color: #00BB00;">9887</span> mapped genes, <span style="color: #00BB00;">491284</span> non-zero values. Likelihood: <span style="color: #00BB00;">Negative Binomial</span>.</span></span>
<span><span class="co">#&gt; → ATAC: <span style="color: #00BB00;">500</span> cells with <span style="color: #00BB00;">56942</span> mapped peaks, <span style="color: #00BB00;">1529825</span> non-zero values. Likelihood: <span style="color: #00BB00;">Negative Binomial</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Clusters: <span style="color: #BB0000;">not available</span>.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="color: #BBBBBB; background-color: #BB00BB; font-weight: bold;"> LOG </span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; - 2023-10-17 11:01:23.266875 Created input object.</span></span>
<span><span class="co">#&gt; - 2023-10-17 11:01:27.083726 Filtered s123egments: [0|50|50]</span></span>
<span><span class="co">#&gt; [1] 0</span></span></code></pre></div>
<!-- ```{r, fig.width=6, fig.height=6} -->
<!-- # Select the first 5 segments to plot -->
<!-- some_segments = get_input(x, what = "segmentation") %>%  -->
<!--   filter(row_number() < 5) %>%  -->
<!--   pull(segment_id) -->
<!-- plot_data(x, what = 'histogram', segments = some_segments) -->
<!-- ``` -->
<!-- ```{r, fig.width=8, fig.height=4} -->
<!-- plot_data(x, what = 'lineplot') -->
<!-- ``` -->
<!-- ```{r, fig.width=6, fig.height=6} -->
<!-- plot_data(x, what = 'heatmap') -->
<!-- ``` -->
<!-- 
# Datasets

## RNA


```r
# Breast-cancer PDX
data('campbell_bcpdx')
campbell_bcpdx %>% print
#> ──  [30m [43m[ (R)CONGAS+ ] [49m [39m [34m Campbell et al. Breast cancer PDX used in CONGAS [39m ─────────────
#> 
#> ── CNA segments (reference:  [32m [32mhg19 [32m [39m)
#> → Input  [32m [32m70 [32m [39m CNA segments, mean ploidy  [32m [32m2.09 [32m [39m.
#> 
#>   [44m  [49m [42m  [49m [42m  [49m [42m  [49m [43m  [49m| [42m  [49m [42m  [49m [44m  [49m| [42m  [49m [42m  [49m [44m  [49m| [42m  [49m [42m  [49m [44m  [49m [42m  [49m| [42m  [49m [44m  [49m| [44m  [49m [43m  [49m| [42m  [49m [42m  [49m| [42m  [49m [47m  [49m [42m  [49m| [44m  [49m [42m  [49m [43m  [49m [42m  [49m| [42m  [49m [42m  [49m [47m  [49m| [42m  [49m| [44m  [49m [42m  [49m [42m  [49m [42m  [49m [44m  [49m [42m  [49m [42m  [49m| [44m  [49m [42m  [49m [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m [44m  [49m [42m  [49m [43m  [49m [43m  [49m [41m  [49m [44m  [49m| [44m  [49m [42m  [49m [43m  [49m [44m  [49m| [42m  [49m [44m  [49m [42m  [49m [44m  [49m| [41m  [49m [43m  [49m [42m  [49m [47m  [49m [41m  [49m| [42m  [49m [44m  [49m| [44m  [49m [43m  [49m| [42m  [49m [42m  [49m
#> 
#>    [4mPloidy: [24m   [46m  [49m 0    [44m  [49m 1    [42m  [49m 2    [43m  [49m 3    [41m  [49m 4    [45m  [49m 5    [47m  [49m *
#> 
#> ── Modalities
#> → RNA:  [32m [32m497 [32m [39m cells with  [32m [32m8246 [32m [39m mapped genes,  [32m [32m1321676 [32m [39m non-zero values. Likelihood:  [32m [32mNegative Binomial [32m [39m.
#> → ATAC:  [31mnot available [39m
#>  [33m! [39m Clusters:  [31mnot available [39m.
#> 
#> ──  [1m [1m [45m [37m LOG  [39m [49m [1m [22m ──
#> 
#> - 2021-04-01 15:09:50 Created input object.
#> - 2021-04-01 15:10:23 Filtered outliers: [14|0.03|0.97]
#> [1] 0
```


```r
# Glioblastoma tumour normal
data('patel_gbmtn')
patel_gbmtn %>% print
#> ──  [30m [43m[ (R)CONGAS+ ] [49m [39m [34m Patel et al. GBM tumour/normal used in CONGAS (SmartSeq) [39m ─────
#> 
#> ── CNA segments (reference:  [32m [32mGRCh38 [32m [39m)
#> → Input  [32m [32m14 [32m [39m CNA segments, mean ploidy  [32m [32m2 [32m [39m.
#> 
#>   [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m [42m  [49m| [42m  [49m [42m  [49m
#> 
#>    [4mPloidy: [24m   [46m  [49m 0    [44m  [49m 1    [42m  [49m 2    [43m  [49m 3    [41m  [49m 4    [45m  [49m 5    [47m  [49m *
#> 
#> ── Modalities
#> → RNA:  [32m [32m75 [32m [39m cells with  [32m [32m1506 [32m [39m mapped genes,  [32m [32m62416 [32m [39m non-zero values. Likelihood:  [32m [32mGaussian [32m [39m.
#> → ATAC:  [31mnot available [39m
#>  [33m! [39m Clusters:  [31mnot available [39m.
#> 
#> ──  [1m [1m [45m [37m LOG  [39m [49m [1m [22m ──
#> 
#> - 2021-04-02 10:32:31 Created input object.
#> [1] 0
```


```r
# Hematological bone marrow failure
data('zaho_hemato')
zaho_hemato %>% print
#> ──  [30m [43m[ (R)CONGAS+ ] [49m [39m [34m Zaho et al. MDS hematopoietical used in CONGAS (HCA) [39m ─────────
#> 
#> ── CNA segments (reference:  [32m [32mGRCh38 [32m [39m)
#> → Input  [32m [32m21 [32m [39m CNA segments, mean ploidy  [32m [32m2 [32m [39m.
#> 
#>   [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m| [42m  [49m
#> 
#>    [4mPloidy: [24m   [46m  [49m 0    [44m  [49m 1    [42m  [49m 2    [43m  [49m 3    [41m  [49m 4    [45m  [49m 5    [47m  [49m *
#> 
#> ── Modalities
#> → RNA:  [32m [32m94 [32m [39m cells with  [32m [32m18841 [32m [39m mapped genes,  [32m [32m890478 [32m [39m non-zero values. Likelihood:  [32m [32mNegative Binomial [32m [39m.
#> → ATAC:  [31mnot available [39m
#>  [33m! [39m Clusters:  [31mnot available [39m.
#> 
#> ──  [1m [1m [45m [37m LOG  [39m [49m [1m [22m ──
#> 
#> - 2021-04-03 12:24:46 Created input object.
#> - 2021-04-03 12:24:48 Filtered outliers: [4.2|0.03|0.97]
#> [1] 0
```
-->
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.caravagnalab.org" class="external-link">Salvatore Milite</a>, <a href="https://www.caravagnalab.org" class="external-link">Riccardo Bergamin</a>, <a href="https://www.caravagnalab.org" class="external-link">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
