<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ Rcongas</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />
<meta property="og:image" content="militeee.github.io/Rcongas/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Rcongas</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/Rcongas.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a></a>
    </li>
    <li>
      <a href="../articles/1_Models.html">Models</a>
    </li>
    <li>
      <a href="../articles/2_Getters.html">Getters</a>
    </li>
    <li>
      <a href="../articles/3_Plotting_functions.html">Plotting functions</a>
    </li>
    <li>
      <a href="../articles/4_Preprocessing.html">Preprocessing</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Old version</li>
    <li>
      <a href="../articles/kieran_vignette.html">Breast PDX analysis</a>
    </li>
    <li>
      <a href="../articles/gbm_vignette.html">Glioblastoma GBM analysis</a>
    </li>
    <li>
      <a href="../articles/monosomy_vignette.html">Monosomy 7 analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/militeee/Rcongas/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-available-data" class="hasAnchor"><a href="#section-available-data" class="anchor"></a>Available data</h2>
          <p class="section-desc"><p>Data released with the package</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="congas_example.html">congas_example</a></code> </p>
        </td>
        <td><p>Example Rcongas fit.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-fitting-functions" class="hasAnchor"><a href="#section-fitting-functions" class="anchor"></a>Fitting functions</h2>
          <p class="section-desc"><p>Functions to cluster the data.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="approx_theta_prior_params.html">approx_theta_prior_params()</a></code> </p>
        </td>
        <td><p>Approximation of the library size prior</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="best_cluster.html">best_cluster()</a></code> </p>
        </td>
        <td><p>Model selection with CONGAS.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="list_models.html">list_models()</a></code> </p>
        </td>
        <td><p>List models available in the package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="recalculate_information_criteria.html">recalculate_information_criteria()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="run_inference.html">run_inference()</a></code> </p>
        </td>
        <td><p>Run CONGAS inference</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-s-functions" class="hasAnchor"><a href="#section-s-functions" class="anchor"></a>S3 functions</h2>
          <p class="section-desc"><p>The <code>rcongas</code> S3 object.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="print.rcongas.html">print(<i>&lt;rcongas&gt;</i>)</a></code> </p>
        </td>
        <td><p>Print for an object of class <code>'rcongas'</code>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="summary.rcongas.html">summary(<i>&lt;rcongas&gt;</i>)</a></code> </p>
        </td>
        <td><p>Summary for an object of class <code>'rcongas'</code>.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot.rcongas.html">plot(<i>&lt;rcongas&gt;</i>)</a></code> </p>
        </td>
        <td><p>Print for an object of class <code>'rcongas'</code>.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-plotting-functions" class="hasAnchor"><a href="#section-plotting-functions" class="anchor"></a>Plotting functions</h2>
          <p class="section-desc"><p>Functions to plot data and fits.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="plot_chromosome.html">plot_chromosome()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_gw_cna_profiles.html">plot_gw_cna_profiles()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_inference_report.html">plot_inference_report()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_counts_rna_segments.html">plot_counts_rna_segments()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_raw_data.html">plot_raw_data()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_highlights.html">plot_highlights()</a></code> </p>
        </td>
        <td><p>plot_highlights = function(x, alpha = 0.05)

  # Best model
  best_model &lt;- Rcongas:::get_best_model(x)
  
  res &lt;-
    data.frame(
      t(best_model$parameters$cnv_probs),
      stringsAsFactors = FALSE,
      check.names = FALSE
    ) <!-- %&gt;% -->
    dplyr::mutate(segment_id = rownames(.)) <!-- %&gt;% -->
    deidify() <!-- %&gt;% -->
    reshape2::melt(
      id.vars = c("chr", "from", "to"),
      variable.name = "cluster",
      value.name = "CN"
    ) <!-- %&gt;% -->
    dplyr::mutate(cluster = as.character(cluster)) <!-- %&gt;% mutate(cluster = ifelse( -->
      str_starts(cluster, pattern = "c|C"),
      cluster,
      paste0("c", cluster)
    )) <!-- %&gt;% -->
    dplyr::as_tibble() <!-- %&gt;% -->
    dplyr::mutate(lognorm_CN = CN)
  
  res$highlights = create_highlights(res, alpha)
  
  # Normalise CNA values for comparisons
  # means = res <!-- %&gt;% -->
  #   dplyr::group_by(chr, from, to) <!-- %&gt;% -->
  #   dplyr::summarise(segment_mean = mean(CN), .groups = 'keep') <!-- %&gt;% -->
  #   dplyr::ungroup()
  #
  # # Bind everything
  # res = res <!-- %&gt;% -->
  #   dplyr::full_join(means, by = c('chr', 'from', 'to')) <!-- %&gt;% -->
  #   dplyr::mutate(lognorm_CN = CN,
  #                 CN = CN - segment_mean)
  
  res$highlight = create_highlights(res, alpha)
  
  # Clones
  clones = Rcongas::get_clusters(x)$cluster <!-- %&gt;% unique %&gt;% sort -->
  
  
  res_split = res <!-- %&gt;% group_split(cluster) -->
  names(res_split) = sapply(res_split, function(x)
    x$cluster[1])
  
  df_density = NULL
  
  for (i in clones)
    for (j in clones)
    
      if (i != j)
      
        f  = get_density_gamma_highlights(res, i, j)
        x_ij = x <!-- %&gt;% -->
          filter(cluster <!-- %in% c(i, j)) %&gt;% -->
          group_by(chr, from, to) <!-- %&gt;% -->
          summarise(diff_ij = abs(diff(lognorm_CN)))
        
        # Fit density function
        den_ij = data.frame(
          x = seq(0, max(x_ij$diff_ij) * 1.25, length.out = 30),
          y = dgamma(
            seq(0, max(x_ij$diff_ij) * 1.25, length.out = 30),
            fit_ij$estimate[[1]],
            fit_ij$estimate[[2]]
          ),
          stringsAsFactors = FALSE
        )
        
        df_density = df_density <!-- %&gt;% -->
          bind_rows(den_ij <!-- %&gt;% mutate(pair = paste0(i, ' vs ', j))) -->
      
    
  
  ggplot(df_density, aes(x=x,y=y, color=pair))+geom_point()</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_latent_variables.html">plot_latent_variables()</a></code> </p>
        </td>
        <td><p>Plot latent varibles.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_mixing_proportions.html">plot_mixing_proportions()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_normalization_factors.html">plot_normalization_factors()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_segment_density.html">plot_segment_density()</a></code> </p>
        </td>
        <td><p>Plots the data and fit density for a set of segments.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="report_analysis.html">report_analysis()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-getters-for-clustering-results" class="hasAnchor"><a href="#section-getters-for-clustering-results" class="anchor"></a>Getters for clustering results</h2>
          <p class="section-desc"><p>Functions to extract clustering information from a model.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_clusters.html">get_clusters()</a></code> </p>
        </td>
        <td><p>Extract cell clustering assignments.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_clusters_ploidy.html">get_clusters_ploidy()</a></code> </p>
        </td>
        <td><p>Extract the clusters ploidy profiles.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_clusters_size.html">get_clusters_size()</a></code> </p>
        </td>
        <td><p>Extract size of clusters.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_k.html">get_k()</a></code> </p>
        </td>
        <td><p>Extract the number of mixture components.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_segment_test_counts.html">get_segment_test_counts()</a></code> </p>
        </td>
        <td><p>Wilcoxon test for difference in RNA counts among groups.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-getters-for-input-data" class="hasAnchor"><a href="#section-getters-for-input-data" class="anchor"></a>Getters for input data</h2>
          <p class="section-desc"><p>Functions to extract information of the data stored in a model.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="get_counts.html">get_counts()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_input_segmentation.html">get_input_segmentation()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_mapped_genes.html">get_mapped_genes()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_input_raw_data.html">get_input_raw_data()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_reference_genome.html">get_reference_genome()</a></code> </p>
        </td>
        <td><p>Reference genome used in a model.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_gene_annotations.html">get_gene_annotations()</a></code> </p>
        </td>
        <td><p>Genome coordinates available in the package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_dataset_stats.html">get_dataset_stats()</a></code> </p>
        </td>
        <td><p>Return dataset statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_segment_ids.html">get_segment_ids()</a></code> </p>
        </td>
        <td><p>Extract the segment ids (for indexing).</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-filter-functions" class="hasAnchor"><a href="#section-filter-functions" class="anchor"></a>Filter functions</h2>
          <p class="section-desc"><p>Functions to filter out information from a model.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="filter_clusters.html">filter_clusters()</a></code> </p>
        </td>
        <td><p>Filter output clusters</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-synthetic-data-generation-functions" class="hasAnchor"><a href="#section-synthetic-data-generation-functions" class="anchor"></a>Synthetic data generation functions</h2>
          <p class="section-desc"><p>Functions to sample from the generative model.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="generate_random_tree.html">generate_random_tree()</a></code> </p>
        </td>
        <td><p>Generate a random tree</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="run_simulation.html">run_simulation()</a></code> </p>
        </td>
        <td><p>Run a complete simulation</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-other-analyses-de-gse-" class="hasAnchor"><a href="#section-other-analyses-de-gse-" class="anchor"></a>Other analyses (DE, GSE)</h2>
          <p class="section-desc"><p>Differential Expression, Gene Set Enrichment, etc.</p></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="calculate_DE.html">calculate_DE()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_DE_table.html">get_DE_table()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_DE_gw.html">plot_DE_gw()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_DE_volcano.html">plot_DE_volcano()</a></code> </p>
        </td>
        <td><p>Title</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://sites.google.com/view/cdslab'>Salvatore Milite</a>, <a href='https://sites.google.com/view/cdslab'>Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


